<div class="container">

<table style="width: 100%;"><tr>
<td>falcon</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Finding Allele-Specific Copy Number in Next-Generation Sequencing Data</h2>

<h3>Description</h3>

<p>This library contains a set of tools for Allele-specific DNA Copy Number Profiling using Next-Generation Sequencing.  Given the allele-specific coverage at a set of variant loci, get getChangepoints program segments the genome into regions of homogeneous allele-specific copy number.  It requires, as input, the read counts supporting each variant allele in each of a pair of case and control samples.  For example, for detection of somatic mutations in a tumors, the case and control samples can be, respectively, the tumor and normal sample from the same individual.
</p>


<h3>Author(s)</h3>

<p>Hao Chen and Nancy Zhang
</p>
<p>Maintainer: Hao Chen (hxchen@ucdavis.edu)
</p>


<h3>References</h3>

<p>Hao Chen, John Bell, Nicolas Zavala, Hanlee Ji and Nancy Zhang. Allele-specific copy number profiling by next-generation DNA sequencing. Nucleic Acids Research, 2014.</p>


<h3>See Also</h3>

<p><code>getChangepoints</code>, <code>getASCN</code>, <code>view</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Example) 
# tauhat = getChangepoints(readMatrix)  # uncomment this to run the function. 
 # This example has 6309 variant loci and it took 94 seconds to run on a laptop with 
 # Intel Core i5-2410M processor.
cn = getASCN(readMatrix, tauhat=tauhat)
 # cn$tauhat would give the indices of change-points.  
 # cn$ascn would give the estimated allele-specific copy numbers for each segment.
 # cn$Haplotype[[i]] would give the estimated haplotype for the major chromosome in segment i 
 # if this segment has different copy numbers on the two homologous chromosomes.
view(cn)
</code></pre>


</div>