<div class="container">

<table style="width: 100%;"><tr>
<td>carrierprob</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Compute mutation carrier probabilities for individuals with missing gentoypes

</h2>

<h3>Description</h3>

<p>Computes model- or data-based carrier probabilities for individuals with missing genotypes based on the observed mutation status of family members and the individual's phenotype.
</p>


<h3>Usage</h3>

<pre><code class="language-R">carrierprob(condition = "geno", method = "data", fit = NULL, data, mode = "dominant", 
q = 0.02)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>condition</code></td>
<td>

<p>Choice of conditional information to use for computing the carrier probability. Possible choices are <code>"geno"</code> for using observed genotypes and <code>"geno+pheno"</code> for using both observed genotype and phenotype information in the calculation of the carrier probability.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>Choice of methods to calculate the carrier probability. Possible choices are <code>"data"</code> for empirical calculation of the carrier probabilities based on data, 
<code>"mendelian"</code> using Mendelian transmission probabilities based on observed carriers within families, or <code>"model"</code> using the parametric model fit; see details below. Default is <code>"data"</code>.
If <code>method = "data"</code>, only <code>data</code> is required to be specified.  
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fit</code></td>
<td>

<p>An object of class <code>penmodel</code>, a fitted model by <code>penmodelEM</code> function for inferring missing mutation statuses in the family.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>

<p>Family data that includes missing genotypes using the same data format generated by the function <code>simfam</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>

<p>Choice of modes of inheritance when using <code>method="model"</code>. Possible choices are <code>"dominant"</code> for dominant model or <code>"recessive"</code> for recessive model. Default is <code>"dominant"</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>

<p>Frequency of the disease causing allele when using <code>method="model"</code>. The value should be between 0 and 1. If <code>NULL</code>, the estimated allele frequency from data will be used. Default value  is <code>0.02</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When <code>method="model"</code> along with the choice of <code>condition="geno+pheno"</code>, the carrier probability for individual <code class="reqn">i</code> is calculated by conditioning on her/his observed phenotype and carrier statuses of family members 
 
P(X<sub>i</sub> = 1 | Y<sub>i</sub> , X<sup>o</sup> )  =
P(Y<sub>i</sub> | X<sub>i</sub> = 1) * P(X<sub>i</sub> = 1 | X<sup>o</sup>) /
(P(Y<sub>i</sub> | X<sub>i</sub> = 1) * P(X<sub>i</sub> = 1 | X<sup>o</sup>) + P(Y<sub>i</sub> | X<sub>i</sub> = 0) * P(X<sub>i</sub> = 0 | X<sup>o</sup>)) 
where X<sub>i</sub> indicates the unknown carrier status of individual <code class="reqn">i</code> and X<sup>o</sup> represents the observed carrier statuses in his or her family members; Y<sub>i</sub> represents the observed phenotype t<sub>i</sub>, δ<sub>i</sub> of individual <code class="reqn">i</code> in terms of age at onset t<sub>i</sub> and disease status indicator δ<sub>i</sub> with 1 used for affected individuals and 0 for unaffected individuals.
When <code>method="mendelian"</code> along with the choice of <code>condition="geno"</code>, the carrier probability is calculated based on Mendelian laws of genetic transmission with a fixed allele frequency.
</p>


<h3>Value</h3>

<p>Returns a data frame with a vector of carrier probabilities called <code>carrp.geno</code> when <code>condition="geno"</code> or <code>carrp.pheno</code> when <code>condtion="geno+pheno"</code> added after the last column of the family data.
</p>


<h3>Author(s)</h3>

<p>Yun-Hee Choi
</p>


<h3>See Also</h3>

<p><code>simfam, penmodelEM, plot.simfam, summary.simfam</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Simulated family data with 30% of members missing their genetic information.

set.seed(4321)
fam &lt;- simfam(N.fam = 100, design = "pop+", base.dist = "Weibull", mrate = 0.3,
       base.parms = c(0.01,3), vbeta = c(-1.13, 2.35), agemin = 20)
 
# EM algorithm for fitting family data with missing genotypes assuming a Weibull
# baseline hazard and dominant mode of Mendelian inheritance for a major gene.

fitEM &lt;- penmodelEM(Surv(time, status) ~ gender + mgene, cluster = "famID", gvar = "mgene", 
        parms = c(0.01, 3, -1.13, 2.35), data = fam, design = "pop+", base.dist = "Weibull", 
        method = "mendelian", mode = "dominant")

# Carrier probability obtained by conditioning on the observed genotypes and phenotype, 
# assuming a dominant Mendelian mode of inheritance

fam.added &lt;- carrierprob(condition = "geno+pheno", method = "model", fit = fitEM, 
             data = fam, mode = "dominant", q = 0.02)

# pedigree plot for family 1 displaying carrier probabilities

plot.simfam(fam.added, famid = 1)

</code></pre>


</div>