<div class="container">

<table style="width: 100%;"><tr>
<td>simulate_poisson_gene_data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate Gene Expression Data from Poisson NMF or Multinomial
Topic Model</h2>

<h3>Description</h3>

<p>Simulate count data from a Poisson NMF model or
multinomial topic model, in which topics represent “gene
expression programs”, and gene expression programs are
characterized by different rates of expression. The way in which
the counts are simulated is modeled after gene expression studies
in which expression is measured by single-cell RNA sequencing
(“RNA-seq”) techniques: each row of the counts matrix
corresponds a gene expression profile, each column corresponds to a
gene, and each matrix element is a “read count”, or
“UMI count”, measuring expression level. Factors are
simulated so as to capture realistic changes in gene expression
across different cell types. See “Details” for the procedure
used to simulate factors, loadings and counts.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulate_poisson_gene_data(n, m, k, s, p = 1, sparse = FALSE)

simulate_multinom_gene_data(n, m, k, sparse = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>Number of rows in the simulated count matrix. Should be at
least 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>Number of columns in the simulated count matrix. Should be
at least 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>Number of factors, or “topics”, used to generate
the data. Should be 2 or more.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>s</code></td>
<td>
<p>Vector of “size factors”; each row of the loadings
matrix <code>L</code> is scaled by the entries of <code>s</code> before
generating the counts. This should be a vector of length n
containing only positive values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Probability that <code>F[i,j]</code> is equal to the mean rate.
Smaller values of <code>p</code> will result in more factors that are the
same across topics.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sparse</code></td>
<td>
<p>If <code>sparse = TRUE</code>, convert the counts matrix to
a sparse matrix in compressed, column-oriented format; see
<code>sparseMatrix</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Here we describe the process for generating the n x k
loadings matrix <code>L</code> and the m x k factors matrix <code>F</code>.
</p>
<p>Each row of the <code>L</code> matrix is generated in the following
manner: (1) the number of nonzero mixture proportions is <code class="reqn">1
  \le n \le k</code>, with probability proportional to <code class="reqn">2^{-n}</code>;
(2) the indices of the nonzero mixture proportions are sampled
uniformly at random; and (3) the nonzero mixture proportions are
sampled from the Dirichlet distribution with <code class="reqn">\alpha = 1</code> (so
that all topics are equally likely).
</p>
<p>Each row of the factors matrix are generated according to the
following procedure: (1) generate <code class="reqn">u = |r| - 5</code>, where <code class="reqn">r ~
  N(0,2)</code>; (2) for each topic <code class="reqn">k</code>, generate the Poisson rates as
<code class="reqn">exp(max(t,-5))</code>, where <code class="reqn">t ~ 0.95 * N(u,s/10) + 0.05 *
  N(u,s)</code>, and <code class="reqn">s = exp(-u/8)</code>. Factors can be interpreted as
Poisson rates or multinomial probabilities, so that individual
counts can be viewed as being generated from a weighted mixture
of “topics” with different rates or probabilities.
</p>
<p>Once the loadings and factors have been generated, the counts are
simulated from either the Poisson NMF or multinomial topic model:
for the former, <code>X[i,j]</code> is Poisson with rate <code>Y[i,j]</code>,
where <code>Y = tcrossprod(L,F)</code>; for the latter, <code>X[i,]</code> is
multinomial with size <code>s[i]</code> and with class probabilities
<code>P[i,]</code>, where <code>P = tcrossprod(L,F)</code>. For the multinomial
model only, the sizes <code>s</code> are randomly generated as <code>s =
  10^rnorm(n,3,0.2)</code>.
</p>
<p>Note that only minimal argument checking is performed;
the function is mainly used to test implementation of the
topic-model-based differential count analysis.
</p>


<h3>Value</h3>

<p><code>simulate_poisson_gene_data</code> returns a list containing
the counts matrix <code>X</code>, and the size factors <code>s</code> and
factorization, <code>F</code>, <code>L</code>, used to generate the counts.
<code>simulate_multinom_gene_data</code> returns a list containing the
counts matrix <code>X</code>, and the mixture proportions <code>L</code> and
factors (gene probabilities, or relative gene expression levels)
<code>F</code> used to generate the counts.
</p>


</div>