<div class="container">

<table style="width: 100%;"><tr>
<td>pbmc_facs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Mixture of 10 FACS-purified PBMC Single-Cell RNA-seq data</h2>

<h3>Description</h3>

<p>These data are a selection of the reference
transcriptome profiles generated via single-cell RNA sequencing
(RNA-seq) of 10 bead-enriched subpopulations of PBMCs (Donor A),
described in Zheng <em>et al</em> (2017). The data are unique
molecular identifier (UMI) counts for 16,791 genes in 3,774 cells.
(Genes with no expression in any of the cells were removed.) Since
the majority of the UMI counts are zero, they are efficiently
stored as a 16,791 x 3774 sparse matrix. These data are used in the
vignette illustrating how ‘fastglmpca’ can be used to
analyze single-cell RNA-seq data. Data for a separate set of 1,000
cells is provided as a “test set” to evaluate out-of-sample
predictions.
</p>


<h3>Format</h3>

<p><code>pbmc_facs</code> is a list with the following elements:
</p>

<dl>
<dt>counts</dt>
<dd>
<p>16,791 x 3,774 sparse matrix of UMI counts, with
rows corresponding to genes and columns corresponding to
cells (samples). It is an object of class <code>"dgCMatrix"</code>).</p>
</dd>
<dt>counts_test</dt>
<dd>
<p>UMI counts for an additional test set of 100
cells.</p>
</dd>
<dt>samples</dt>
<dd>
<p>Data frame containing information about the
samples, including cell barcode and source FACS population
(“celltype” and “facs_subpop”).</p>
</dd>
<dt>samples_test</dt>
<dd>
<p>Sample information for the additional test
set of 100 cells.</p>
</dd>
<dt>genes</dt>
<dd>
<p>Data frame containing information and the genes,
including gene symbol and Ensembl identifier.</p>
</dd>
<dt>fit</dt>
<dd>
<p>GLM-PCA model that was fit to the UMI count data in
the vignette.</p>
</dd>
</dl>
<h3>Source</h3>

<p><a href="https://www.10xgenomics.com/resources/datasets">https://www.10xgenomics.com/resources/datasets</a>
</p>


<h3>References</h3>

<p>G. X. Y. Zheng <em>et al</em> (2017). Massively parallel digital
transcriptional profiling of single cells. <em>Nature Communications</em>
<b>8</b>, 14049. <a href="https://doi.org/10.1038/ncomms14049">doi:10.1038/ncomms14049</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library(Matrix)
data(pbmc_facs)
cat(sprintf("Number of genes: %d\n",nrow(pbmc_facs$counts)))
cat(sprintf("Number of cells: %d\n",ncol(pbmc_facs$counts)))
cat(sprintf("Proportion of counts that are non-zero: %0.1f%%.\n",
            100*mean(pbmc_facs$counts &gt; 0)))

</code></pre>


</div>