<div class="container">

<table style="width: 100%;"><tr>
<td>run_homer</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform HOMER Motif Enrichment Analysis using DE Genomic Positions</h2>

<h3>Description</h3>

<p>Run HOMER motif finding algorithm
(<code>findMotifsGenome.pl</code>) to identify motifs enriched for
differentially expressed (DE) genomic positions. See
<a href="http://homer.ucsd.edu">http://homer.ucsd.edu</a> for more information.
</p>


<h3>Usage</h3>

<pre><code class="language-R">run_homer(
  de,
  k,
  positions,
  genome = "hg19",
  subset = function(postmean, lpval, lfsr, rank, quantile) lfsr &lt; 0.05,
  homer.exec = "findMotifsGenome.pl",
  out.dir = tempdir(),
  homer.options = "-len 8,10,12 -size 200 -mis 2 -S 25 -p 1 -h",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>de</code></td>
<td>
<p>An object of class “topic_model_de_analysis”,
usually the result of running <code>de_analysis</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>Use the DE analysis results for this topic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>positions</code></td>
<td>
<p>A table of genomic positions corresponding to rows
of the <code>de_analysis</code> results. Specifically, it should a data
frame with four columns: “chr”, chromosome name or number;
“start”, start position of genomic feature; “end”,
end position of genomic feature; and “name”, the name of the
genomic feature. If not specified, the genomic positions will be
extracted from the row names of <code>de$postmean</code>, in which the
row names are expected to be of the form <code>chr_start_end</code>. The
genomic positions will be written to a BED file (see
<a href="https://genome.ucsc.edu/FAQ/FAQformat.html">https://genome.ucsc.edu/FAQ/FAQformat.html</a> for more
information about BED files).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome</code></td>
<td>
<p>The genome parameter passed to
<code>findMotifsGenome.pl</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset</code></td>
<td>
<p>Describe input argument "subset" here.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>homer.exec</code></td>
<td>
<p>The name or file path of the HOMER
<code>findMotifsGenome.pl</code> excutable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out.dir</code></td>
<td>
<p>The positions BED file and HOMER results are written
to this directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>homer.options</code></td>
<td>
<p>Character string used to override default
<code>findMotifsGenome.pl</code> options.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>When <code>verbose = TRUE</code>, progress information is
printed to the console.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame containing the motif enrichment results. It
is created from the <code>knownResults.txt</code> HOMER output.
</p>


<h3>References</h3>

<p>Heinz, S., Benner, C., Spann, N., Bertolino, E., Lin, Y. C., Laslo,
P., Cheng, J. X., Murre, C., Singh, H. and Glass, C. K. (2010).
Simple combinations of lineage-determining transcription factors
prime cis-regulatory elements required for macrophage and B cell
identities. <em>Molecular Cell</em> <b>38</b>, 576-589.
</p>


</div>