<div class="container">

<table style="width: 100%;"><tr>
<td>binary_cfdr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform cFDR leveraging binary auxiliary covariates</h2>

<h3>Description</h3>

<p>Perform cFDR leveraging binary auxiliary covariates
</p>


<h3>Usage</h3>

<pre><code class="language-R">binary_cfdr(p, q, group)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>p-values for principal trait (vector of length n)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>
<p>binary auxiliary data values (vector of length n)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>group membership of each SNP for leave-one-out procedure (vector of length n) (e.g. chromosome number or LD block)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>data.frame of p, q and v values
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing functional genomic data).
# We use the parameters_in_locfdr() function to extract the parameters estimated by
# the locfdr function.

# generate p
set.seed(2)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
q &lt;- rbinom(n, 1, 0.1)

group &lt;- c(rep("A", n/2), rep("B", n/2)) 

binary_cfdr(p, q, group)

</code></pre>


</div>