<div class="container">

<table style="width: 100%;"><tr>
<td>parameters_in_locfdr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>parameters_in_locfdr</h2>

<h3>Description</h3>

<p>parameters_in_locfdr
</p>


<h3>Usage</h3>

<pre><code class="language-R">parameters_in_locfdr(
  p,
  q,
  indep_index,
  res_p = 300,
  res_q = 500,
  maf = NULL,
  check_indep_cor = TRUE,
  enforce_p_q_cor = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>p values for principal trait (vector of length n)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>q</code></td>
<td>
<p>continuous auxiliary data values (vector of length n)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indep_index</code></td>
<td>
<p>indices of independent SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>res_p</code></td>
<td>
<p>resolution for p</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>res_q</code></td>
<td>
<p>resolution for q</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>minor allele frequencies for SNPs to which <code>p</code> and <code>q</code> relate (optional and used to perform MAF matching)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check_indep_cor</code></td>
<td>
<p>check that sign of the correlation between <code>p</code> and <code>q</code> is the same in the independent subset as in the whole</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>enforce_p_q_cor</code></td>
<td>
<p>if <code>p</code> and <code>q</code> are negatively correlated, flip the sign on <code>q</code> values</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>list of values used as input into <code>locfdr::locfdr</code> function intrinsically in <code>flexible_cfdr</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# In this example, we generate some p-values (representing GWAS p-values)
# and some arbitrary auxiliary data values (e.g. representing functional genomic data).
# We use the parameters_in_locfdr() function to extract the parameters estimated by
# the locfdr function.

# generate p
set.seed(1)
n &lt;- 1000
n1p &lt;- 50 
zp &lt;- c(rnorm(n1p, sd=5), rnorm(n-n1p, sd=1))
p &lt;- 2*pnorm(-abs(zp))

# generate q
mixture_comp1 &lt;- function(x) rnorm(x, mean = -0.5, sd = 0.5)
mixture_comp2 &lt;- function(x) rnorm(x, mean = 2, sd = 1)
q &lt;- c(mixture_comp1(n1p), mixture_comp2(n-n1p))

n_indep &lt;- n

parameters_in_locfdr(p, q, indep_index = 1:n_indep)


</code></pre>


</div>