<div class="container">

<table style="width: 100%;"><tr>
<td>SOTAclustering</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>SOTA Clustering</h2>

<h3>Description</h3>

<p>Self-organizing Tree Algorithm (SOTA) introduced by [Herrero et al., 2001].
</p>


<h3>Usage</h3>

<pre><code class="language-R">SOTAclustering(Data, ClusterNo,PlotIt=FALSE,UnrestGrowth,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Data</code></td>
<td>
<p>[1:n,1:d] matrix of dataset to be clustered. It consists of n cases of d-dimensional data points. Every case has d attributes, variables or features.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ClusterNo</code></td>
<td>
<p>A number k which defines k different clusters to be built by the algorithm.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PlotIt</code></td>
<td>
<p>Default: FALSE, if TRUE plots the first three dimensions of the dataset with colored three-dimensional data points defined by the clustering stored in <code>Cls</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>UnrestGrowth</code></td>
<td>

<p>TRUE: forces the <code>ClusterNo</code> option to uphold.
FALSE: enables the algorithm to find its own number of clusters, in this cases ClusterNo should contain a high number because it is internally set as the number of iterations which is either reached or the max diversity criteria is satisfied priorly.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further arguments to be set for the clustering algorithm, if not set, default arguments are used.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List of
</p>
<table>
<tr style="vertical-align: top;">
<td><code>Cls</code></td>
<td>
<p>[1:n]  numerical vector with n numbers defining the classification as the main output of the clustering algorithm. It has k unique numbers representing the arbitrary labels of the clustering.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sotaObject</code></td>
<td>
<p>Object defined by clustering algorithm as the other output of this algorithm</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>*Luis Winckelman intergrated several function from clValid because it's ORPHANED.
</p>


<h3>Author(s)</h3>

<p>Luis Winckelmann*, Vasyl Pihur, Guy Brock, Susmita Datta, Somnath Datta</p>


<h3>References</h3>

<p>[Herrero et al., 2001]  Herrero, J., Valencia, A., &amp; Dopazo, J.: A hierarchical unsupervised growing neural network for clustering gene expression patterns, Bioinformatics, Vol. 17(2), pp. 126-136. 2001.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#Does Work
data('Hepta')
out=SOTAclustering(Hepta$Data,ClusterNo=7)
table(Hepta$Cls,out$Cls)


#Does not work well
data('Lsun3D')
out=SOTAclustering(Lsun3D$Data,ClusterNo=100,PlotIt=FALSE,UnrestGrowth=FALSE)

</code></pre>


</div>