<div class="container">

<table style="width: 100%;"><tr>
<td>fcrosFCmat</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculation of a matrix of fold changes using pairwise comparisons</h2>

<h3>Description</h3>

<p>This function is used internally by fcros2() to compute a matrix 
of fold changes using pairwise comparisons of a two biological conditions dataset.</p>


<h3>Usage</h3>

<pre><code class="language-R">fcrosFCmat(xdata, cont, test, log2.opt=0, trim.opt=0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xdata</code></td>
<td>
<p> A table containing a two biological conditions
dataset to process for obtaining a matrix of fold changes
that results from pairwise comparisons of samples.
The rownames of xdata are used for the output idnames.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cont</code></td>
<td>
<p> A vector containing label names of the control samples:
<code>cont</code> = c("cont01", "cont02", ...)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>
<p> A vector containing label names of the test samples:
<code>test</code> = c("test01", "test02", "test03", ...)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
the data values in matrix "xdata" are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values of each gene
are not used for computing the statistic "ri" or "u1", i.e. the
interquartile rank values are averaged:
<code>trim.opt</code> = 0.25</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The label names appearing in the parameters "cont" and "test" should match
some label names of the columns in the table "xdata". The dataset "xdata" can
contain other label names not used.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 3 components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fcMat </code></td>
<td>
<p>A matrix of fold changes associated with genes. This matrix is
obtained using pairwise comparisons of samples
in the dataset "xdata".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FC </code></td>
<td>
<p>The fold changes for genes in the dataset "xdata". These
fold changes are calculated as a ratio of averages from the
test and the control samples. Non log scale values are used
in this calculation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FC2 </code></td>
<td>
<p>The robust fold changes for genes in the dataset "xdata". These
fold changes are calculated as a trimmed mean of the fold
changes obtained using pairs of samples. Non log scale values
are used in this calculation.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed 
genes, BMC Bioinformatics, 2014, 15:14<br></p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class="language-R">   data(fdata);

   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcrosFCmat()
   fc &lt;- fcrosFCmat(fdata, cont, test, log2.opt);

   # plot histogram of the fold change (log2 scale) in dataset
   hist(log2(fc$FC),nclass = 50);
</code></pre>


</div>