<div class="container">

<table style="width: 100%;"><tr>
<td>fcros-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Method to Search for Differentially Expressed Genes and to Detect Recurrent
Chromosomal Copy Number Aberrations</h2>

<h3>Description</h3>

<p>Implementation of a method based on fold change rank ordering
statistics to search for differentially expressed genes or to
detect recurrent chromosomal copy number aberrations. This package
can be used for two biological conditions high-throughput dataset 
(microarray, RNA-seq, ...), for expression profiling dataset over time
without replicates or for cytogenetics dataset (aCGH, Sequencing).
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
        Package: </td>
<td style="text-align: left;"> fcros</td>
</tr>
<tr>
<td style="text-align: left;">
        Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
        Version: </td>
<td style="text-align: left;"> 1.6.1</td>
</tr>
<tr>
<td style="text-align: left;">
        Date: </td>
<td style="text-align: left;"> 2019-05-28</td>
</tr>
<tr>
<td style="text-align: left;">
        License: </td>
<td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
<td style="text-align: left;">
    </td>
</tr>
</table>
<p>Package fcros has the following functions:
</p>

<table>
<tr>
<td style="text-align: left;">
         fcros(): </td>
<td style="text-align: left;"> The function to use with a dataset from two biological condition samples. The</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  dataset should be in a single table. The function fcros() performs a pairwise</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  conparison of samples to obtain a matrix of fold changes. The fold changes</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  are sorted and their rank values are used to associate statistic with genes/probes.</td>
</tr>
<tr>
<td style="text-align: left;">
         fcros2(): </td>
<td style="text-align: left;"> The function to use with datasets from two biological biological conditions. The</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  datasets should be in two separate tables as inputs. The</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  function fcros2() performs a pairwise comparison of samples from each table</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> to
                  obtain fold changes. The fold changes are sorted, their rank values are</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                   combined and then used to associate statistic with genes/probes.</td>
</tr>
<tr>
<td style="text-align: left;">
         pfco(): </td>
<td style="text-align: left;"> The function to use with a dataset from two biological condition samples. The</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  dataset should be in a single table. The function pfco() performs a pairwise</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  conparison of samples to obtain a matrix of fold changes. The fold changes</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  are sorted and their rank values are used to associate statistic with genes/probes</td>
</tr>
<tr>
<td style="text-align: left;">
                  </td>
<td style="text-align: left;"> using a singular value decomposition.</td>
</tr>
<tr>
<td style="text-align: left;">
         fcrosMod(): </td>
<td style="text-align: left;"> This function uses fold changes or ratios matrix as input to associate
                  statistic</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> with genes/probes.</td>
</tr>
<tr>
<td style="text-align: left;">
         pfcoMod(): </td>
<td style="text-align: left;"> This function uses fold changes or ratios matrix as input to associate
                  statistic</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> with genes/probes using a singular value decomposition.</td>
</tr>
<tr>
<td style="text-align: left;">
         fcrosFCmat(): </td>
<td style="text-align: left;"> This function allows to compute a matrix of fold changes using</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                   pairwise comparisons of the two biological condition samples in a dataset.</td>
</tr>
<tr>
<td style="text-align: left;">
         fcrosTtest(): </td>
<td style="text-align: left;"> This function allows to use the Student t-test to calculate p-values</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  for the genes in a dataset.</td>
</tr>
<tr>
<td style="text-align: left;">
         fcrosRead(): </td>
<td style="text-align: left;"> This function allows to read a tab delimited text file to be use as an</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> 
                  input for the function fcros(), fcros2() or fcrosMod().</td>
</tr>
<tr>
<td style="text-align: left;">
         fcrosWrite(): </td>
<td style="text-align: left;"> This function allows to save the results obtained using the function
                  fcros(),</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> fcros2() or fcrosMod() in a tab delimited text file.</td>
</tr>
<tr>
<td style="text-align: left;">
         fcrosTopN(): </td>
<td style="text-align: left;"> This function allows to search for the top N down- and/or up-regulated genes
                  </td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> from the results obtained using the function fcros(), fcros2(), pfco(),</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  fcrosMod() or pfcoMod().</td>
</tr>
<tr>
<td style="text-align: left;">
         fvalTopN(): </td>
<td style="text-align: left;"> This function allows to search for the top N down- and/or up-regulated genes
                  </td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> from the results obtained using the function fcros(), fcros2(), pfco(),</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  fcrosMod() or pfcoMod().</td>
</tr>
<tr>
<td style="text-align: left;">
         pvalTopN(): </td>
<td style="text-align: left;"> This function allows to search for the top N down- and/or up-regulated genes
                  </td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> from the results obtained using the function fcros(), fcros2(), pfco(),</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  fcrosMod() or pfcoMod().</td>
</tr>
<tr>
<td style="text-align: left;">
         histoPlot(): </td>
<td style="text-align: left;"> This function plots on the screen the histogram of the FCROS statistics
                  </td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> obtained using the results of the function fcros(), fcros2(), pfco(),</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  fcrosMod() or pfcoMod()</td>
</tr>
<tr>
<td style="text-align: left;">
         fvalVolcanoPlot(): </td>
<td style="text-align: left;"> This function performs a volcano plot of the results obtained</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  using the function fcros(), fcros2(), pfco(), fcrosMod() or pfcomod()</td>
</tr>
<tr>
<td style="text-align: left;">
         pvalVolcanoPlot(): </td>
<td style="text-align: left;"> This function performs a volcano plot of the results obtained</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  using the function fcros(), fcros2(), pfco(), fcrosMod() or pfcoMod()</td>
</tr>
<tr>
<td style="text-align: left;">
         chrSummary(): </td>
<td style="text-align: left;"> This function summarizes detection results by chromosome</td>
</tr>
<tr>
<td style="text-align: left;">
         chrSegment(): </td>
<td style="text-align: left;"> This function segments a chromosome data</td>
</tr>
<tr>
<td style="text-align: left;">
         chrPlot(): </td>
<td style="text-align: left;"> This function performs a plot of the chromosome probes data</td>
</tr>
<tr>
<td style="text-align: left;">
         chrPlot2(): </td>
<td style="text-align: left;"> This function performs a plot of the chromosome segmentation results</td>
</tr>
<tr>
<td style="text-align: left;">
         voomReads(): </td>
<td style="text-align: left;"> This function performs a transformation of the read counts</td>
</tr>
<tr>
<td style="text-align: left;">
         tcnReads(): </td>
<td style="text-align: left;"> This function performs a total count normalization of reads</td>
</tr>
<tr>
<td style="text-align: left;">
         rankReads(): </td>
<td style="text-align: left;"> The function to use with a dataset from two biological condition samples. 
                  The</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> dataset should be in a single table. The function rankReads() performs a</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> 
                  pairwise conparison of samples to obtain a matrix of fold changes. Small uniform</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  values are added to read counts. This is repeated nrun time. The fold changes</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;">
                  are sorted and their rank values are used to associate statistic with genes/probes.</td>
</tr>
<tr>
<td style="text-align: left;">
         scoreThr(): </td>
<td style="text-align: left;"> Using the log10 transformed score values obtained with the rankReads(),
                  this function</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> computes numerically the inflection point value given lower
                  and upper bound</td>
</tr>
<tr>
<td style="text-align: left;"></td>
<td style="text-align: left;"> values for the slope region.
    </td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Doulaye Dembele
Maintainer: Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed 
genes, BMC Bioinformatics, 2014, 15:14<br></p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462<br></p>
<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class="language-R">   data(fdata);

   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt, trim.opt);

   # perform Volcano plot
   fvalVolcanoPlot(af, thr = 0.01)

   # save fcros values in a file
   fcrosWrite(af, file = "test2delete_values.txt");

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(c(0,11), ncol = 1);
   id.up &lt;- matrix(c(rep(0,11)), ncol = 1);
   n &lt;- length(af$FC);
   f.value &lt;- af$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown+1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup+1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);

   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1,2:21], type = "l", col = "blue", xlim = c(1,20),
          ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t, data.down[i,2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20),
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>


</div>