<div class="container">

<table style="width: 100%;"><tr>
<td>calcSRmatMod</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculation of the sorted rank matrix from the dataset</h2>

<h3>Description</h3>

<p>This is an internal function used to calculate the sorted
rank matrix. It is used in the functions: fcrosMod() and pfcoMod().</p>


<h3>Usage</h3>

<pre><code class="language-R">calcSRmatMod(xdata, samp, log2.opt=0, trim.opt=0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xdata</code></td>
<td>
<p> A matrix containing fold changes or ratios from a biological
dataset to process for searching differentially expressed
genes or for detecting recurrent copy number aberrations 
regions: <code>fcMat</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>samp</code></td>
<td>
<p> A vector of sample label names which should appear in the columns
of the matrix fcMat: <code>samp</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
values in the matrix "fcMat" are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values for each gene 
are not used for computing the statistic "ri", i.e. the 
interquartile range rank values are averaged:
<code>trim.opt</code> = 0.25</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kaster P, Fold change rank ordering statistics:
a new method for detecting differentially expressed 
genes, BMC bioinformatics, 2014, 15:14</p>


<h3>Examples</h3>

<pre><code class="language-R">#    data(fdata);
</code></pre>


</div>