<div class="container">

<table style="width: 100%;"><tr>
<td>pvalTopN</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search for the top N changed genes or probes using p-values</h2>

<h3>Description</h3>

<p>This function allows to seach for the top N 
differentially expressed genes or changed probes.
It uses the p-value. These p-values can be obtained using one of
the following functions fcros(), fcros2(), fcrosMod(), pfco(),
pfcoMod() or another statistical method.</p>


<h3>Usage</h3>

<pre><code class="language-R">pvalTopN(pval, topN)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pval</code></td>
<td>
<p> This is vector with p-values obtained using the functions fcros(),
fcros2(), fcrosMod(), pfco(), pfcoMod() or another statistical method:
<code>pval = af$p.value</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>topN</code></td>
<td>
<p> The expected number of the top DE genes/probes in the dataset used:
<code>topN</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p> This function returns a data frame containing 2 components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>alpha </code></td>
<td>
<p>Two threshold values for the down- and the up-regulated allowing
to have the top N DE genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>index </code></td>
<td>
<p>The indexes of the top N DE genes</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes,
BMC Bioinformatics, 2014, 15:14<br></p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class="language-R">   data(fdata);

   rownames(fdata) &lt;- fdata[,1]
   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt);
   
   # now select top 10 down and/or up regulated genes
   top10 &lt;- pvalTopN(af$p.value, 12);

   # display thresholds
   top10$alpha
   
   # display index of top10 genes
   fdata[top10$index, 1]
   
   # display fvalue of the top10 genes
   (af$f.value)[top10$index]
</code></pre>


</div>