<div class="container">

<table style="width: 100%;"><tr>
<td>fcros2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Search for differentially expressed genes/probes</h2>

<h3>Description</h3>

<p>Implementation of a method based on fold change or ratio rank
ordering statistics for detecting differentially expressed 
genes. This function should be used with dataset in two separate
tables and from two biological conditions datasets (microarray,
RNA-seq, ...). Pairwise combinations of samples from
the two biological conditions are used to obtain a matrix of
fold changes. For each combination, the FCs obtained are
sorted in an increasing order and the corresponding rank values are
associated with genes/probes. 
Then, a statistic is associated with each gene/probe.</p>


<h3>Usage</h3>

<pre><code class="language-R">fcros2(xdata1, xdata2, cont, test, log2.opt = 0, trim.opt = 0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xdata1</code></td>
<td>
<p> A matrix or a table containing two biological conditions
dataset to process for detecting differentially expressed
genes. The rownames of xdata1 are used for the output idnames.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xdata2</code></td>
<td>
<p> A matrix or a table containing two biological conditions
dataset to process fordetecting differentially expressed
genes: <code>xdata2</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cont</code></td>
<td>
<p> A vector containing the label names of the control samples:
<code>cont</code> = c("cont01", "cont02", ...)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>
<p> A vector containing the label names of the test samples:
<code>test</code> = c("test01", "test02", "test03", ...)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
data in the tables "xdata1" and "xdata2" are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and  the upper rank values for a gene
are not used for computing  the statistics "ri", i.e. the 
interquartile range
rank values are averaged: <code>trim.opt</code> = 0.25</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The label names appearing in the parameters "cont" and "test" should 
match some label names in the columns of the data matrices "xdata1" and "xdata2". 
It is not necessary to use all column label names appearing in matrices "xdata1" 
and "xdata2". However, it is assumed that the same genes (same IDs or symbol) are 
used in xdata1 and xdata2.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 9 components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>idnames</code></td>
<td>
<p> A vector containing the list of the IDs or symbols associated with genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ri </code></td>
<td>
<p>The average of rank values associated with genes in the
datasets. These values are rank statistics leading to f-values
and p-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FC </code></td>
<td>
<p>The fold changes for genes. These fold changes are
calculated as a ratio of averages from the test and control
samples. Non log scale values are used in the calculation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FC2 </code></td>
<td>
<p>The robust fold changes for genes. These fold changes are
calculated as a trimmed mean of the fold changes or ratios
obtained from the dataset samples using the pairwise comparisons.
Non log scale values are used in the calculation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f.value </code></td>
<td>
<p>The f-values are probabilities associated with genes. These 
values are obtained using the "mean" and the
"standard deviation" ("sd") of the statistics "ri". The "mean" and
"sd" are used as a normal distribution parameters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value </code></td>
<td>
<p>The p-values associated with genes. These values are obtained
from the f-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bounds </code></td>
<td>
<p>Two values which are the lower and the upper bound or the
minimum and the maximum values of non standardized "ri".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>params </code></td>
<td>
<p>Three values, which are the estimates for the parameters "delta"
(average difference between consecutive ordered average of rank) 
"mean" (mean value of variable "ri") and the standard deviation ("sd")
of variable "ri".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>params_t </code></td>
<td>
<p>Three values, which are the theoretical levels for
parameters "delta", "mean" and "sd".</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes, 
BMC Bioinformatics, 2014, 15:14<br></p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class="language-R">   data(fdata);

   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont02", "cont03", "cont04", "cont05",
             "cont06", "cont07", "cont08", "cont09", "cont10");
   test &lt;- c("test01", "test02", "test03", "test04", "test05",
             "test06", "test07", "test08", "test09", "test10");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform fcros2()
   xdata1 &lt;- fdata[,c(2:5, 12:17)];
   xdata2 &lt;- fdata[,c(6:11, 18:21)];
   rownames(xdata1) &lt;- fdata[,1];
   rownames(xdata2) &lt;- fdata[,1];

   af2 &lt;- fcros2(xdata1, xdata2, cont, test, log2.opt, trim.opt);

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af2, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(0,1);
   id.up &lt;- matrix(0,1);
   n &lt;- length(af2$FC);
   f.value &lt;- af2$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown + 1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup + 1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);

   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1,2:21], type = "l", col = "blue", xlim = c(1,20),
        ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t,data.down[i,2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20),
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>


</div>