<div class="container">

<table style="width: 100%;"><tr>
<td>pfco</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Searching for differentially expressed genes/probes using an
approach based on the Perron-Frobenius theorem</h2>

<h3>Description</h3>

<p>Implementation of a method based on fold change and the Perron
theorem for detecting differentially expressed genes in a
dataset.
This function should be used with two biological conditions
dataset (microarray or RNA-seq, ...).
Using pairwise combinations of samples from the
two biological conditions, fold changes (FC) are calculated.
For each combination, the FC obtained are sorted in increasing
order and corresponding rank values are associated with genes.
Then, a statistic is assigned to the robust average ordered rank
values for each gene/probe.</p>


<h3>Usage</h3>

<pre><code class="language-R">pfco(xdata, cont, test, log2.opt = 0, trim.opt = 0.25)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xdata</code></td>
<td>
<p> A matrix or a table containing two biological conditions
dataset to process for detecting differentially expressed
genes. The rownames of xdata are used for the output idnames.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cont</code></td>
<td>
<p> A vector containing the label names of the control samples:
<code>cont</code> = c("cont01", "cont02", ...).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>test</code></td>
<td>
<p> A vector containing the label names of the test samples:
<code>test</code> = c("test01", "test02", "test03", ...).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log2.opt</code></td>
<td>
<p> A scalar equals to 0 or 1. The value 0 (default) means that
data in the matrix "xdata" are expressed in a log2 scale:
<code>log2.opt</code> = 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim.opt</code></td>
<td>
<p> A scalar between 0 and 0.5. The value 0.25 (default) means
that 25% of the lower and the upper rank values of each gene are not
used for computing its statistics "ri", i.e. the interquartile range
rank values are averaged: <code>trim.opt</code> = 0.25</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Label names appearing in the parameter "samp" should match
with some label names in the columns of the data matrix "xdata". It is not 
necessary to use all label names appearing in the columns of the dataset matrix.</p>


<h3>Value</h3>

<p> This function returns a data frame containing 9 components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>idnames</code></td>
<td>
<p> A vector containing the list of IDs or symbols associated with genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ri </code></td>
<td>
<p>The average of rank values associated with genes.
These values are rank values statistics leading to f-values
and p-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FC </code></td>
<td>
<p>The fold changes for genes in the dataset. These fold changes are
calculated as a ratio of averages from the test and the control
samples. Non log scale values are used in the calculation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>FC2 </code></td>
<td>
<p>The robust fold changes for genes. These fold changes are
calculated as a trimmed mean of the fold changes or ratios
obtained from the dataset samples.  Non log scale values are used 
in the calculation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f.value </code></td>
<td>
<p>The f-values are probabilities associated with genes using
the "mean" and the "standard deviation" ("sd") of the statistics "ri".
The "mean" and "sd" are used as a normal distribution parameters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value </code></td>
<td>
<p>The p-values associated with genes. These values are obtained
using a one sample Student t-test on the fold change rank values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comp </code></td>
<td>
<p>Singular values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comp.w </code></td>
<td>
<p>Singular values weights.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comp.wcum </code></td>
<td>
<p>Cumulative sum of the singular values weights.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D, Analysis of high biological data using their rank
values, Stat Methods Med Res, accepted for publication, 2018</p>


<h3>Examples</h3>

<pre><code class="language-R">   data(fdata);
   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;
   trim.opt &lt;- 0.25;

   # perform pfco()
   af &lt;- pfco(fdata, cont, test, log2.opt, trim.opt);

   # now select top 20 down and/or up regulated genes
   top20 &lt;- fcrosTopN(af, 20);
   alpha1 &lt;- top20$alpha[1];
   alpha2 &lt;- top20$alpha[2];
   id.down  &lt;- matrix(0, 1);
   id.up &lt;- matrix(0, 1);
   n &lt;- length(af$FC);
   f.value &lt;- af$f.value;

   idown &lt;- 1;
   iup &lt;- 1;
   for (i in 1:n) {
       if (f.value[i] &lt;= alpha1) { id.down[idown] &lt;- i; idown &lt;- idown + 1; }
       if (f.value[i] &gt;= alpha2) { id.up[iup] &lt;- i; iup &lt;- iup + 1; }
   }

   data.down &lt;- fdata[id.down[1:(idown-1)], ];
   ndown &lt;- nrow(data.down);
   data.up &lt;- fdata[id.up[1:(iup-1)], ];
   nup &lt;- nrow(data.up);


   # now plot down regulated genes
   t &lt;- 1:20;
   op = par(mfrow = c(2,1));
   plot(t, data.down[1,2:21], type = "l", col = "blue", xlim = c(1,20), 
        ylim = c(0,18), main = "Top down-regulated genes");
   for (i in 2:ndown) {
       lines(t,data.down[i,2:21], type = "l", col = "blue")
   }

   # now plot down and up regulated genes
   plot(t, data.up[1,2:21], type = "l", col = "red", xlim = c(1,20), 
       ylim = c(0,18), main = "Top up-regulated genes");
   for (i in 2:nup) {
       lines(t, data.up[i,2:21], type = "l", col = "red")
   }
   par(op)
</code></pre>


</div>