<div class="container">

<table style="width: 100%;"><tr>
<td>histoPlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Histogram plot function of the fcros package results</h2>

<h3>Description</h3>

<p>This function allows to have a histogram plot. It uses the 
statistics "ri" or "u1" obtained using one of the following functions: fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod().</p>


<h3>Usage</h3>

<pre><code class="language-R">histoPlot(af, nbins = 50)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>af</code></td>
<td>
<p> This is an object obtained using the function fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod()</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nbins</code></td>
<td>
<p> This parameter is used for the number of bins in
the histogram. Default setting is 50: <code>nbins = 50</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p> This function plots a histogram on the screen.</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics:
a new method for detecting differentially expressed genes,
BMC Bioinformatics, 2014, 15:14<br></p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class="language-R">   data(fdata);
   rownames(fdata) &lt;- fdata[,1];

   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcros() and pfco()
   af &lt;- fcros(fdata, cont, test, log2.opt);
   af2 &lt;- pfco(fdata, cont, test, log2.opt);

   # Histogram plots
   op &lt;- par(mfrow = c(1,2))
      histoPlot(af);
      histoPlot(af2);
   par(op);
</code></pre>


</div>