<div class="container">

<table style="width: 100%;"><tr>
<td>pvalVolcanoPlot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Performs a volcano plot of the FCROS/PFCO statistics</h2>

<h3>Description</h3>

<p>This function allows to have a volcano like plot using the output 
results of the function fcros(), fcros2(), fcrosMod(), pfco() or pfcomod():
p-values versus robust fold changes (FC2). 
The p-value are transformed using -log10(), while FC2 are transformed using log2().</p>


<h3>Usage</h3>

<pre><code class="language-R">pvalVolcanoPlot(af, thr = 0.05)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>af</code></td>
<td>
<p> This is an object obtained using the functions fcros(),
fcros2(), fcrosMod(), pfco() or pfcoMod():
<code>af = fcros(xdata, cont, test)</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thr</code></td>
<td>
<p> The threshold to obtain the DE genes in the dataset (red plots):
<code>thr = 0.05</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p> This function displays on the screen a volcano like plot using the p-values
and the robust fold changes (FC2).</p>


<h3>Author(s)</h3>

<p>Doulaye Dembele doulaye@igbmc.fr</p>


<h3>References</h3>

<p>Dembele D and Kastner P, Fold change rank ordering statistics: 
a new method for detecting differentially expressed genes,
BMC bioinformatics, 2014, 15:14<br></p>
<p>Dembele D and Kastner P, Comment on: Fold change rank ordering statistics:
a new method for detecting differentially expressed
genes, BMC Bioinformatics, 2016, 17:462</p>


<h3>Examples</h3>

<pre><code class="language-R">   data(fdata);

   rownames(fdata) &lt;- fdata[,1]
   cont &lt;- c("cont01", "cont07", "cont03", "cont04", "cont08");
   test &lt;- c("test01", "test02", "test08", "test09", "test05");
   log2.opt &lt;- 0;

   # perform fcros()
   af &lt;- fcros(fdata, cont, test, log2.opt);
   
   # Volcano plot
   pvalVolcanoPlot(af, thr = 1e-6);
</code></pre>


</div>