<div class="container">

<table style="width: 100%;"><tr>
<td>average.power.hart</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute average power for RNA-seq experiments assuming Negative Binomial distribution</h2>

<h3>Description</h3>

<p>Compute average power for RNA-seq experiments assuming Negative Binomial distribution
</p>


<h3>Usage</h3>

<pre><code class="language-R">average.power.hart(n, alpha, log.fc, mu, sig)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.fc</code></td>
<td>
<p>log fold-change (vector), usual null hypothesis is log.fc=0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>read depth per gene (vector, same length as log.fc)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sig</code></td>
<td>
<p>coefficient of variation (CV) per gene (vector, same length as log.fc)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The power function is based on equation (1) of Hart et al (2013). It assumes a Negative Binomial model for RNA-seq read counts and equal sample size per group.
</p>


<h3>Value</h3>

<p>Average power estimate for multiple testing procedure
</p>


<h3>References</h3>

<p>SN Hart, TM Therneau, Y Zhang, GA Poland, and J-P Kocher (2013). Calculating Sample Size Estimates for RNA Sequencing Data. Journal of Computational Biology 20: 970-978.
</p>


<h3>See Also</h3>

<p><code>power.hart</code> for more details about power calculation of data under Negative Binomial distribution. The power calculation is based on asymptotic normal approximation.
</p>


<h3>Examples</h3>

<pre><code class="language-R">logFC = log(rep(c(1,2),c(900,100)));
mu = rep(5,1000);
sig = rep(0.6,1000);
average.power.hart(n = 50, alpha = 0.05,log.fc = logFC, mu = mu, sig = sig)
</code></pre>


</div>