<div class="container">

<table style="width: 100%;"><tr>
<td>power.hart</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compute power for RNA-seq experiments assuming Negative Binomial distribution</h2>

<h3>Description</h3>

<p>Use the formula of Hart et al (2013) to compute power for comparing RNA-seq expression across two groups assuming a Negative Binomial distribution. The power calculation is based on asymptotic normal approximation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">power.hart(n, alpha, log.fc, mu, sig)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>per-group sample size (scalar)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>p-value threshold (scalar)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log.fc</code></td>
<td>
<p>log fold-change (vector), usual null hypothesis is log.fc=0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mu</code></td>
<td>
<p>read depth per gene (vector, same length as log.fc)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sig</code></td>
<td>
<p>coefficient of variation (CV) per gene (vector, same length as log.fc)</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is based on equation (1) of Hart et al (2013). It assumes a Negative Binomial model for RNA-seq read counts and equal sample size per group.
</p>


<h3>Value</h3>

<p>Vector of power estimates for the set of two-sided tests
</p>


<h3>References</h3>

<p>SN Hart, TM Therneau, Y Zhang, GA Poland, and J-P Kocher (2013). Calculating Sample Size Estimates for RNA Sequencing Data. Journal of Computational Biology 20: 970-978.
</p>


<h3>Examples</h3>

<pre><code class="language-R">n.hart = 2*(qnorm(0.975)+qnorm(0.9))^2*(1/20+0.6^2)/(log(2)^2)   # Equation (6) of Hart et al
power.hart(n.hart,0.05,log(2),20,0.6)                            # Recapitulate 90% power
</code></pre>


</div>