<div class="container">

<table style="width: 100%;"><tr>
<td>qiime2meco</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transform 'QIIME2' results to 'microtable' object.</h2>

<h3>Description</h3>

<p>Transform 'QIIME2' qza results to microtable object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">qiime2meco(
  feature_table,
  sample_table = NULL,
  match_table = NULL,
  taxonomy_table = NULL,
  phylo_tree = NULL,
  rep_fasta = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>feature_table</code></td>
<td>
<p>the ASV abundance data with qza format, such as the <code>'data2_table.qza'</code> in the example.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample_table</code></td>
<td>
<p>default NULL; the sample metadata table; four types of formats are available: <br>
1) q2-type tab seperated file of QIIME2, such as the <code>'sample-metadata.tsv'</code> in the example;<br>
2) comma seperated file with the suffix csv or tab seperated file with suffix tsv or txt;<br>
3) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed;<br>
4) <code>data.frame</code> object from R.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>match_table</code></td>
<td>
<p>default NULL; a two column table used to replace the sample names in feature table; Must be two columns without column names;
The first column must be raw sample names same with those in feature table, 
the second column must be new sample names same with the rownames in sample_table; Please also see the example files.
If provided, must be one of the several types of formats: <br>
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv or txt; <br>
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; <br>
3) <code>data.frame</code> object from R.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxonomy_table</code></td>
<td>
<p>default NULL; the taxonomy assignment data with qza format, such as the <code>'taxonomy.qza'</code> in the example.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phylo_tree</code></td>
<td>
<p>default NULL; the phylogenetic tree with qza format, such as the <code>'tree.qza'</code> in the example.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rep_fasta</code></td>
<td>
<p>default NULL; the representative sequences with qza format, such as the <code>'dada2_rep_set.qza'</code> in the example.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>parameter passed to <code>microtable$new</code> function of <code>microeco</code> package, such as <code>auto_tidy</code> parameter.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>microtable</code> object.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# The data files is downloaded from https://docs.qiime2.org/2020.8/tutorials/pd-mice/ 
#   and stored inside the package.
abund_file_path &lt;- system.file("extdata", "dada2_table.qza", package="file2meco")
sample_file_path &lt;- system.file("extdata", "sample-metadata.tsv", package="file2meco")
taxonomy_file_path &lt;- system.file("extdata", "taxonomy.qza", package="file2meco")
qiime2meco(abund_file_path)
qiime2meco(abund_file_path, sample_table = sample_file_path)
qiime2meco(abund_file_path, sample_table = sample_file_path, 
  taxonomy_table = taxonomy_file_path)

## End(Not run)
</code></pre>


</div>