<div class="container">

<table style="width: 100%;"><tr>
<td>qiime1meco</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transform 'QIIME' results to 'microtable' object.</h2>

<h3>Description</h3>

<p>Transform 'QIIME' results to microtable object. 
The QIIME results refer in particular to the files of qiime1 software.
</p>


<h3>Usage</h3>

<pre><code class="language-R">qiime1meco(
  feature_table,
  sample_table = NULL,
  match_table = NULL,
  phylo_tree = NULL,
  rep_fasta = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>feature_table</code></td>
<td>
<p>the otu table generated from 'QIIME'. Taxonomic information should be in the end of the file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample_table</code></td>
<td>
<p>default NULL; sample metadata table; If provided, must be one of the several types of formats: 
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv/txt; 
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; 
3) <code>data.frame</code> object from R.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>match_table</code></td>
<td>
<p>default NULL; a two column table used to replace the sample names in feature table; Must be two columns without column names;
The first column must be raw sample names same with those in feature table, 
the second column must be new sample names same with the rownames in sample_table; Please also see the example files.
If provided, must be one of the several types of formats: <br>
1) comma seperated file with the suffix csv or tab seperated file with suffix tsv/txt; <br>
2) Excel type file with the suffix xlsx or xls; require <code>readxl</code> package to be installed; <br>
3) <code>data.frame</code> object from R.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phylo_tree</code></td>
<td>
<p>default NULL; the phylogenetic tree; generally, a file with suffix "tre".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rep_fasta</code></td>
<td>
<p>default NULL; the representative sequences; a fasta file, generally with suffix "fasta" or "fna" or "fa".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>parameter passed to microtable$new function of microeco package, such as <code>auto_tidy</code> parameter.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>microtable</code> object.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# use the raw data files stored inside the package
otu_file_path &lt;- system.file("extdata", "otu_table_raw.txt", package="file2meco")
sample_file_path &lt;- system.file("extdata", "sample_info.csv", package="file2meco")
phylo_file_path &lt;- system.file("extdata", "rep_phylo.tre", package="file2meco")
rep_fasta_path &lt;- system.file("extdata", "rep.fna", package="file2meco")
qiime1meco(otu_file_path, sample_table = sample_file_path)
qiime1meco(otu_file_path, sample_table = sample_file_path, 
  phylo_tree = phylo_file_path)
qiime1meco(otu_file_path, sample_table = sample_file_path, 
  phylo_tree = phylo_file_path, rep_fasta = rep_fasta_path)

## End(Not run)
</code></pre>


</div>