<div class="container">

<table style="width: 100%;"><tr>
<td>EBFST</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Empirical Bayes estimator of Fst.</h2>

<h3>Description</h3>

<p>This function estimates global/pairwise Fst among subpopulations using empirical Bayes method (Kitada et al. 2007, 2017). Preciseness of estimated pairwise Fst is evaluated by bootstrap method. This function accepts two types of data object, GENEPOP data (Rousset 2008) and allele (haplotype) frequency data (Kitada et al. 2007). Missing genotype values in the GENEPOP file ("0000" or "000000") are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class="language-R">EBFST(popdata, num.iter = 100, locus = F)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>popdata</code></td>
<td>
<p>Genotype data object of populations created by read.genepop function from a GENEPOP file. Allele (haplotype) frequency data object created by read.frequency function from a frequency format file also is acceptable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.iter</code></td>
<td>
<p>A positive integer value specifying the number of iterations in empirical Bayes simulation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locus</code></td>
<td>
<p>A Logical argument indicating whether locus-specific Fst values should be calculated.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Frequency format file is a plain text file containing allele (haplotype) count data. This format is mainly for mitochondrial DNA (mtDNA) haplotype frequency data, however nuclear DNA (nDNA) data also is applicable. In the data object created by read.frequency function, "number of samples" means haplotype count. Therefore, it equals the number of individuals in mtDNA data, however it is the twice of the number of individuals in nDNA data. First part of the frequency format file is the number of subpopulations, second part is the number of loci, and latter parts are [<em>population</em> x <em>allele</em>] matrices of the observed allele (haplotype) counts at each locus. Two examples of frequency format files are attached in this package. See <code>jsmackerel</code>.
</p>


<h3>Value</h3>

<p>global:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>Estimated gene flow rate.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fst</code></td>
<td>
<p>Estimated genome-wide global Fst.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fst.locus</code></td>
<td>
<p>Estimated locus-specific global Fst. (If locus = TRUE)</p>
</td>
</tr>
</table>
<p>pairwise:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>fst</code></td>
<td>
<p>Estimated genome-wide pairwise Fst.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fst.boot</code></td>
<td>
<p>Bootstrap mean of estimated Fst.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fst.boot.sd</code></td>
<td>
<p>Bootstrap standard deviation of estimated Fst.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fst.locus</code></td>
<td>
<p>Estimated locus-specific pairwise Fst. (If locus = TRUE)</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Hirohisa Kishino, Shuichi Kitada</p>


<h3>References</h3>

<p>Kitada S, Kitakado T, Kishino H (2007) Empirical Bayes inference of pairwise FST and its distribution in the genome. <em>Genetics</em>, 177, 861-873.<br><br>
Kitada S, Nakamichi R, Kishino H (2017) The empirical Bayes estimators of fine-scale population structure in high gene flow species. <em>Mol. Ecol. Resources</em>, DOI: 10.1111/1755-0998.12663 <br><br>
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code>read.genepop</code>, <code>read.frequency</code>,
<code>as.dist</code>, <code>as.dendrogram</code>,
<code>hclust</code>, <code>cmdscale</code>, <code>nj</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.genepop(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# Fst estimation
result.eb &lt;- EBFST(popdata)
ebfst &lt;- result.eb$pairwise$fst
ebfst.d &lt;- as.dist(ebfst)
print(ebfst.d)

# dendrogram
ebfst.hc &lt;- hclust(ebfst.d,method="average")
plot(as.dendrogram(ebfst.hc), xlab="",ylab="",main="", las=1)

# MDS plot
mds &lt;- cmdscale(ebfst.d)
plot(mds, type="n", xlab="",ylab="")
text(mds[,1],mds[,2], popdata$pop_names)

# NJ tree
library(ape)
ebfst.nj &lt;- nj(ebfst.d)
plot(ebfst.nj,type="u",main="",sub="")
</code></pre>


</div>