<div class="container">

<table style="width: 100%;"><tr>
<td>pop_pairwiseFST</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genome-wide poppulation-paiwise FST (Nei &amp; Chesser 1983).</h2>

<h3>Description</h3>

<p>This function estimates genome-wide poppulation-paiwise FST among subpopulations based on Nei and Chesser's corrected GST (Nei&amp;Chesser 1983) from a GENEPOP data object (Rousset 2008). Missing genotype values in the GENEPOP file ("0000" or "000000") are simply ignored.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pop_pairwiseFST(popdata)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>popdata</code></td>
<td>
<p>Population data object created by <code>read.GENEPOP</code> function from a GENEPOP file.</p>
</td>
</tr></table>
<h3>Value</h3>

<p>Estimated genome-wide population-pairwise FST.</p>


<h3>Author(s)</h3>

<p>Reiichiro Nakamichi, Shuichi Kitada, Hirohisa Kishino</p>


<h3>References</h3>

<p>Nei M, Chesser RK (1983) Estimation of fixation indices and gene diversity. <em>Annals of Human Genetics</em>, 47, 253-259.<br><br>
Rousset F (2008) Genepop'007: a complete reimplementation of the Genepop software for Windows and Linux. <em>Mol. Ecol. Resources</em>, 8, 103-106.
</p>


<h3>See Also</h3>

<p><code>read.GENEPOP</code>,
<code>as.dist</code>, <code>as.dendrogram</code>,
<code>hclust</code>, <code>cmdscale</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Example of GENEPOP file
data(jsmackerel)
jsm.ms.genepop.file &lt;- tempfile()
jsm.popname.file &lt;- tempfile()
cat(jsmackerel$MS.genepop, file=jsm.ms.genepop.file, sep="\n")
cat(jsmackerel$popname, file=jsm.popname.file, sep=" ")

# Data load
# Prepare your GENEPOP file and population name file in the working directory.
# Replace "jsm.ms.genepop.file" and "jsm.popname.file" by your file names.
popdata &lt;- read.GENEPOP(genepop=jsm.ms.genepop.file, popname=jsm.popname.file)

# GST estimation
result.poppairFST &lt;- pop_pairwiseFST(popdata)
poppairFST.d &lt;- as.dist(result.poppairFST)
print(poppairFST.d)

# dendrogram
poppairFST.hc &lt;- hclust(poppairFST.d,method="average")
plot(as.dendrogram(poppairFST.hc), xlab="",ylab="",main="", las=1)

# MDS plot
mds &lt;- cmdscale(poppairFST.d)
plot(mds, type="n", xlab="",ylab="")
text(mds[,1],mds[,2], popdata$pop_names)
</code></pre>


</div>