<div class="container">

<table style="width: 100%;"><tr>
<td>sample_data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Preprocessed 293Tâ€“Jurkat scRNA-seq dataset</h2>

<h3>Description</h3>

<p>This data set has been generated from 293T and Jurkat cell data containing a total of ~3200 cells,
with an almost equal number of representative transcriptomes of each type.
The cells were mixed in vitro at equal proportions.
Authors of the study resolved the cell types bioinformatically exploiting their SNV profiles.
With this data, we mimicked the rare cell phenomenon by bioinformatically diluting
Jurkat cell proportion to 2.5% in the data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(sample_data)</code></pre>


<h3>Format</h3>

<p>A matrix containing 1580 observations (in rows), with 1000 features per observation (in columns).</p>


<h3>Source</h3>

<p>10xgenomics, https://support.10xgenomics.com/single-cell-gene-expression/datasets</p>


<h3>References</h3>

<p>Zheng, G. X. et al.
Massively parallel digital transcriptional profiling of single cells.
<em>Nat. Commun.</em> 8, 14049 (2017).
</p>


</div>