<div class="container">

<table style="width: 100%;"><tr>
<td>fishtree_alignment</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get aligned sequences from the Fish Tree of Life</h2>

<h3>Description</h3>

<p>Retrieves an aligned sequence via the Fish Tree of Life API. If neither 'species' nor 'rank' are specified, returns the entire sequence matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fishtree_alignment(species, rank, split = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p>(Optionally) subset the results based on a vector of species names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rank</code></td>
<td>
<p>(Optionally) subset the results based on the supplied taxonomic rank.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split</code></td>
<td>
<p>Splits the output into a list by gene locus.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class '"DNAbin"', or a named list of the same if 'split = TRUE“
</p>


<h3>References</h3>

<p>Rabosky, D. L., Chang, J., Title, P. O., Cowman, P. F., Sallan, L., Friedman, M., Kashner, K., Garilao, C., Near, T. J., Coll, M., Alfaro, M. E. (2018). An inverse latitudinal gradient in speciation rate for marine fishes. Nature, 559(7714), 392–395. doi:10.1038/s41586-018-0273-1
</p>


<h3>See Also</h3>

<p>DNAbin
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
surgeon_dna &lt;- fishtree_alignment(rank = "Acanthuridae", split = TRUE)
surgeon_dna[[1]]
par(mfrow = c(9, 3), mar = c(0.5, 0.5, 1, 0.5), xaxt = "n", yaxt = "n")
for (gene in names(surgeon_dna)) {
  image(surgeon_dna[[gene]], legend = FALSE, show.labels = FALSE)
  title(gene)
}

## End(Not run)
</code></pre>


</div>