<div class="container">

<table style="width: 100%;"><tr>
<td>prediction.errors</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Computes the prediction errors using the trained models.</h2>

<h3>Description</h3>

<p>Computes the prediction errors using the trained models.
</p>


<h3>Usage</h3>

<pre><code class="language-R">prediction.errors(models, expression, attribute, weather)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>models</code></td>
<td>
<p>A list of trained models for the genes of interest.
</p>
<p>At the moment the collection of trained models returned
by <code>FIT::train()</code> cannot be directly passed to <code>FIT::predict()</code>:
the user has to explicitly convert it to an appropriate format by using
<code>FIT::train.to.predict.adaptor()</code>.
(This restriction might be removed in a future.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>expression</code></td>
<td>
<p>An object that represents the actual measured data of
gene expressions.
The object can be created from a dumped/saved dataframe
of size <code>nsamples * ngenes</code>
using <code>FIT::load.expression()</code>.
(At the moment it is an instance of a hidden class IO$Attribute,
but this may be subject to change.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>attribute</code></td>
<td>
<p>An object that represents the attributes of
microarray/RNA-seq data.
The object can be created from a dumped/saved dataframe
using <code>FIT::load.attribute()</code>.
(At the moment it is an instance of a hidden class IO$Attribute,
but this may be subject to change.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weather</code></td>
<td>
<p>An object that represents actual or hypothetical weather data
with which predictions of gene expressions are made.
The object can be created from a dumped/saved dataframe
using <code>FIT::load.weather()</code>.
(At the moment it is an instance of a hidden class IO$Weather,
but this may be subject to change.)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of deviance (a measure of validity of predictions,
as is defined by each model) between the prediction results
and the measured results (as is provided by the user through
<code>expression</code> argument).
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# see the usage of FIT::predict()

## End(Not run)
</code></pre>


</div>