<div class="container">

<table style="width: 100%;"><tr>
<td>gtex</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GTEx data</h2>

<h3>Description</h3>

<p>Derived from data made available by the Genotype Tissue
Expression (GTEx) project (Lonsdale et al. 2013),
which provides <code class="reqn">z</code>-scores for assessing the significance of effects of
genetic variants (single nucleotide polymorphisms, or SNPs) on gene
expression across 44 human tissues. To reduce the data to a more manageable
size, Urbut et al. (2019) chose the "top" SNP for
each gene — that is, the SNP associated with the largest (absolute)
<code class="reqn">z</code>-score over all 44 tissues. This yields a <code class="reqn">16,069 \times 44</code> matrix
of <code class="reqn">z</code>-scores, with rows corresponding to SNP-gene pairs and columns
corresponding to tissues. The dataset included here
is further subsampled down to 1000 rows.
</p>


<h3>Format</h3>

<p><code>gtex</code> is a matrix with 1000 rows and 44 columns, with rows
corresponding to SNP-gene pairs and columns corresponding to tissues.
</p>


<h3>Source</h3>

<p>&lt;https://github.com/stephenslab/gtexresults/blob/master/data/MatrixEQTLSumStats.Portable.Z.rds&gt;
</p>


<h3>References</h3>

<p>Lonsdale et al. (2013).
"The Genotype-Tissue Expression (GTEx) project." <em>Nature Genetics</em>
45(6), 580–585.
</p>
<p>Urbut, Wang, Carbonetto, and Stephens (2019).
"Flexible statistical methods for estimating and testing effects in
genomic studies with multiple conditions." <em>Nature Genetics</em>
51(1), 187–195.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(gtex)
summary(gtex)

</code></pre>


</div>