<div class="container">

<table style="width: 100%;"><tr>
<td>fluidigmAnalysisWrapper</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run the Fluidigm Analysis Script Together</h2>

<h3>Description</h3>

<p>This function serves as a wrapper for the entire analysis pipeline. It takes a Fluidigm input file and performs several operations including conversion to 'PLINK' format, error estimation, calculation of pairwise similarities, determination of pairwise similarity loci, and calculation of the similarity matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fluidigmAnalysisWrapper(
  file,
  out = NA,
  outdir = NA,
  db = NA,
  appendSamplesToDB = FALSE,
  map = NA,
  keep.rep = 1,
  neg_controls = NA,
  y.marker = NA,
  x.marker = NA,
  sp.marker = NA,
  plots = TRUE,
  allele_error = 5,
  marker_dropout = 15,
  no_marker = 50,
  male.y = 3,
  male.hetX = 0,
  female.y = 0,
  female.Xtot = 8,
  female.hetXtot = 3,
  warning.noYtot = 2,
  warning.noHetXtot = 3,
  rearrange = TRUE,
  group = NA,
  fixNames = TRUE,
  sexing = TRUE,
  similarity = 0.85,
  verbose = TRUE,
  verbosity = 1,
  missing.geno = "0 0",
  overwrite = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>A string specifying the path to the Fluidigm input file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out</code></td>
<td>
<p>A string specifying the output file name. If left empty, the original basename of the input file will be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outdir</code></td>
<td>
<p>A string specifying the output folder. If left empty the original folder path of the input file will be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>A string specifying the filepath to the database file. If not provided, the function will proceed with the existing data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>appendSamplesToDB</code></td>
<td>
<p>A logical indicating whether new samples should be added to the database. Default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map</code></td>
<td>
<p>A string specifying the filepath to the PlateDnoY.map file. If not provided, the function will use the map file with the same name as the ped file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.rep</code></td>
<td>
<p>A numeric value indicating the number of replicates to keep. Default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>neg_controls</code></td>
<td>
<p>A vector specifying the names of negative controls. Default is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y.marker</code></td>
<td>
<p>A vector specifying the Y markers for sexing. Default is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x.marker</code></td>
<td>
<p>A vector specifying the X markers for sexing. Default is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sp.marker</code></td>
<td>
<p>A vector specifying the markers used for species identification. Default is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plots</code></td>
<td>
<p>A logical indicating whether plots should be created. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allele_error</code></td>
<td>
<p>A numeric value specifying the threshold for RERUN on Allele errors. Default is 5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker_dropout</code></td>
<td>
<p>A numeric value specifying the threshold for RERUN on Marker dropout. Default is 15.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no_marker</code></td>
<td>
<p>A numeric value specifying the number of markers. Default is 50.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>male.y</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, male y-chromosome markers. Default is 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>male.hetX</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, heterozygote x-chr markers. Default is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>female.y</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, female y-chromosome markers. Default is 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>female.Xtot</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, total female x-chr markers. Default is 8.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>female.hetXtot</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, heterozygote x-chr markers. Default is 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warning.noYtot</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, when should warning be triggered. Default is 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warning.noHetXtot</code></td>
<td>
<p>A numeric value specifying the threshold for sexing, when should warning be triggered. Default is 3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rearrange</code></td>
<td>
<p>A logical indicating whether the ped/map output should be rearranged in order of provided map file. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>A string specifying the sample identifier for statistics. Default is NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fixNames</code></td>
<td>
<p>A logical indicating whether whitespaces from sample names should be automatically removed. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sexing</code></td>
<td>
<p>A logical indicating whether sexing should be performed. Default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>similarity</code></td>
<td>
<p>Similarity threshold. Default: 0.85.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical or numerical value indicating whether the output should be verbose. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbosity</code></td>
<td>
<p>A numerical value indicating the level of verbosity. Set to a higher number for more details. Default is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missing.geno</code></td>
<td>
<p>A character string specifying how missing values should be coded. Default is "0 0".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overwrite</code></td>
<td>
<p>A logical indicating wheter the original map file should be overwritten or not. Default FALSE</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function first checks the input parameters and sets default values if necessary. It then runs the following functions in order:
</p>

<ul>
<li>
<p> fluidigm2PLINK: Converts the Fluidigm data to PLINK format.
</p>
</li>
<li>
<p> estimateErrors: Estimates errors in the PLINK ped files.
</p>
</li>
<li>
<p> calculatePairwiseSimilarities: Calculates pairwise similarities between genotypes.
</p>
</li>
<li>
<p> getPairwiseSimilarityLoci: Determines the loci of pairwise similarities.
</p>
</li>
<li>
<p> similarityMatrix: Calculates the similarity matrix.
The function prints a completion message when all operations are done.
</p>
</li>
</ul>
<h3>Value</h3>

<p>A list containing the following elements:
gensim, a matrix indicating if genotypes are called correctly for replicates and/or if genotypes are missing
summs, a matrix with summary statistics
</p>


<h3>References</h3>


<ul>
<li>
<p>Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D, Maller J, Sklar P, de Bakker PIW, Daly MJ &amp; Sham PC (2007)
<em>PLINK: a toolset for whole-genome association and population-based linkage analysis.</em> American Journal of Human Genetics, 81.
</p>
</li>
<li>
<p>Purcell, Shaun. PLINK. <a href="https://zzz.bwh.harvard.edu/plink/">https://zzz.bwh.harvard.edu/plink/</a></p>
</li>
</ul>
<h3>See Also</h3>


<ul>
<li>
<p><code>fluidigm2PLINK</code>: Converts the Fluidigm data to PLINK format.
</p>
</li>
<li>
<p><code>estimateErrors</code>: Estimates errors in the PLINK ped files.
</p>
</li>
<li>
<p><code>calculatePairwiseSimilarities</code>: Calculates pairwise similarities between genotypes.
</p>
</li>
<li>
<p><code>getPairwiseSimilarityLoci</code>: Determines the loci of pairwise similarities.
</p>
</li>
<li>
<p><code>similarityMatrix</code>: Calculates the similarity matrix.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
  fluidigmAnalysisWrapper(file="path/to/your/file.csv", map="path/to/your/mapfile.map")

## End(Not run)

</code></pre>


</div>