<div class="container">

<table style="width: 100%;"><tr>
<td>exclusionPower</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Power of exclusion</h2>

<h3>Description</h3>

<p>Computes the power (of a single marker, or for a collection of markers) of
excluding a claimed relationship, given the true relationship.
</p>


<h3>Usage</h3>

<pre><code class="language-R">exclusionPower(
  claimPed,
  truePed,
  ids,
  markers = NULL,
  source = "claim",
  disableMutations = NA,
  exactMaxL = Inf,
  nsim = 1000,
  seed = NULL,
  alleles = NULL,
  afreq = NULL,
  knownGenotypes = NULL,
  Xchrom = FALSE,
  plot = FALSE,
  plotMarkers = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>claimPed</code></td>
<td>
<p>A <code>ped</code> object (or a list of such), describing the claimed
relationship. If a list, the sets of ID labels must be disjoint, that is,
all ID labels must be unique.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>truePed</code></td>
<td>
<p>A <code>ped</code> object (or a list of such), describing the true
relationship. ID labels must be consistent with <code>claimPed</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>Individuals available for genotyping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markers</code></td>
<td>
<p>A vector indicating the names or indices of markers attached
to the source pedigree. If NULL (default), then all markers attached to the
source pedigree are used. If <code>alleles</code> or <code>afreq</code> is non-NULL, then this
parameter is ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>source</code></td>
<td>
<p>Either "claim" (default) or "true", deciding which pedigree is
used as source for marker data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>disableMutations</code></td>
<td>
<p>This parameter determines how mutation models are
treated. Possible values are as follows:
</p>

<ul>
<li> <p><code>NA</code> (the default): Mutations are disabled only for those markers whose
known genotypes are compatible with both <code>claimPed</code> and <code>truePed</code>. This is
determined by temporarily removing all mutation models and checking which
markers have nonzero likelihood in both alternatives.
</p>
</li>
<li> <p><code>TRUE</code>: Mutations are disabled for all markers.
</p>
</li>
<li> <p><code>FALSE</code>: No action is done to disable mutations.
</p>
</li>
<li>
<p> A vector containing the names or indices of those markers for which
mutations should be disabled.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exactMaxL</code></td>
<td>
<p>A positive integer, or <code>Inf</code> (default). Exact EPs are
calculated for markers whose number of alleles is less or equal to
<code>exactMaxL</code>; remaining markers are handled by simulation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsim</code></td>
<td>
<p>A positive integer; the number of simulations used for markers
whose number of alleles exceeds <code>exactMaxL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>An integer seed for the random number generator (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles, afreq, Xchrom</code></td>
<td>
<p>If these are given, they are used (together with
<code>knownGenotypes</code>) to create a marker object on the fly.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>knownGenotypes</code></td>
<td>
<p>A list of triplets <code style="white-space: pre;">⁠(a, b, c)⁠</code>, indicating that
individual <code>a</code> has genotype <code>b/c</code>. Ignored unless <code>alleles</code> or <code>afreq</code> is
non-NULL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>
<p>Either a logical or the character "plotOnly". If the latter, a
plot is drawn, but no further computations are done.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plotMarkers</code></td>
<td>
<p>A vector of marker names or indices whose genotypes are to
be included in the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function implements the formula for exclusion power as defined and
discussed in (Egeland et al., 2014).
</p>
<p>It should be noted that <code>claimPed</code> and <code>truePed</code> may be any (lists of)
pedigrees, as long as they both contain the individuals specified by <code>ids</code>.
In particular, either alternative may have inbred founders (with the same or
different coefficients), but this must be set individually for each.
</p>


<h3>Value</h3>

<p>If <code>plot = "plotOnly"</code>, the function returns NULL after producing the
plot.
</p>
<p>Otherwise, the function returns an <code>EPresult</code> object, which is essentially
a list with the following entries:
</p>

<ul>
<li> <p><code>EPperMarker</code>: A numeric vector containing the exclusion power of each
marker. If the known genotypes of a marker are incompatible with the true
pedigree, the corresponding entry is <code>NA</code>.
</p>
</li>
<li> <p><code>EPtotal</code>: The total exclusion power, computed as <code>1 - prod(1 -   EPperMarker, na.rm = TRUE)</code>.
</p>
</li>
<li> <p><code>expectedMismatch</code>: The expected number of markers giving exclusion,
computed as <code>sum(EPperMarker, na.rm = TRUE)</code>.
</p>
</li>
<li> <p><code>distribMismatch</code>: The probability distribution of the number of markers
giving exclusion. This is given as a numeric vector of length <code>n+1</code>, where
<code>n</code> is the number of nonzero elements of <code>EPperMarker</code>. The vector has
names <code>0:n</code>.
</p>
</li>
<li> <p><code>time</code>: The total computation time.
</p>
</li>
<li> <p><code>params</code>: A list containing the (processed) parameters <code>ids</code>, <code>markers</code>
and <code>disableMutations</code>.
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>References</h3>

<p>T. Egeland, N. Pinto and M.D. Vigeland, <em>A general approach to
power calculation for relationship testing.</em> Forensic Science
International: Genetics 9 (2014): 186-190.
<a href="https://doi.org/10.1016/j.fsigen.2013.05.001">doi:10.1016/j.fsigen.2013.05.001</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
############################################
### A standard case paternity case:
### Compute the power of exclusion when the claimed father is in fact
### unrelated to the child.
############################################

# Claim: 'AF' is the father of 'CH'
claim = nuclearPed(father = "AF", children = "CH")

# Attach two (empty) markers
claim = claim |&gt;
  addMarker(alleles = 1:2) |&gt;
  addMarker(alleles = 1:3)

# Truth: 'AF' and 'CH' are unrelated
true = singletons(c("AF", "CH"))

# EP when both are available for genotyping
exclusionPower(claim, true, ids = c("AF", "CH"))

# EP when the child is typed; homozygous 1/1 at both markers
claim2 = claim |&gt;
  setGenotype(marker = 1:2, id = "CH", geno = "1/1")

exclusionPower(claim2, true, ids = "AF")


############################################
### Two females claim to be mother and daughter, but are in reality sisters.
### We compute the power of various markers to reject the claim.
############################################

ids = c("A", "B")
claim = nuclearPed(father = "NN", mother = "A", children = "B", sex = 2)
true = nuclearPed(children = ids, sex = 2)

# SNP with MAF = 0.1:
PE1 = exclusionPower(claimPed = claim, truePed = true, ids = ids,
                     alleles = 1:2, afreq = c(0.9, 0.1))

stopifnot(round(PE1$EPtotal, 5) == 0.00405)

# Tetra-allelic marker with one major allele:
PE2 = exclusionPower(claimPed = claim, truePed = true, ids = ids,
                     alleles = 1:4, afreq = c(0.7, 0.1, 0.1, 0.1))

stopifnot(round(PE2$EPtotal, 5) == 0.03090)


### How does the power change if the true pedigree is inbred?
trueLOOP = halfSibPed(sex2 = 2) |&gt; addChildren(4, 5, ids = ids)

# SNP with MAF = 0.1:
PE3 = exclusionPower(claimPed = claim, truePed = trueLOOP, ids = ids,
                     alleles = 1:2, afreq = c(0.9, 0.1))

# Power almost doubled compared with PE1
stopifnot(round(PE3$EPtotal, 5) == 0.00765)

</code></pre>


</div>