<div class="container">

<table style="width: 100%;"><tr>
<td>simpleSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Unconditional marker simulation</h2>

<h3>Description</h3>

<p>Unconditional simulation of unlinked markers
</p>


<h3>Usage</h3>

<pre><code class="language-R">simpleSim(
  x,
  N,
  alleles,
  afreq,
  ids,
  Xchrom = FALSE,
  mutmod = NULL,
  seed = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>A positive integer: the number of markers to be simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>A vector with allele labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afreq</code></td>
<td>
<p>A numeric vector of allele frequencies. If missing, the alleles
are assumed to be equifrequent.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>A vector containing ID labels of those pedigree members whose
genotypes should be simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Xchrom</code></td>
<td>
<p>A logical: X linked markers or not?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutmod</code></td>
<td>
<p>A list of mutation matrices named 'female' and 'male'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>An integer seed for the random number generator (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Simple genotype simulation, performed by first distributing alleles randomly
to all founders, followed by Mendelian <em>gene dropping</em> down throughout the
pedigree (i.e., for each non-founder a random allele is selected from each of
the parents). Finally, genotypes of individuals not included in <code>ids</code> are
removed.
</p>


<h3>Value</h3>

<p>A <code>ped</code> object equal to <code>x</code> except its <code>MARKERS</code> entry, which
consists of the <code>N</code> simulated markers.
</p>


<h3>See Also</h3>

<p><code>markerSim()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">x = nuclearPed(1)
simpleSim(x, N = 3, afreq = c(0.5, 0.5))

y = cousinPed(1, child = TRUE)
simpleSim(y, N = 3, alleles = LETTERS[1:10])

</code></pre>


</div>