<div class="container">

<table style="width: 100%;"><tr>
<td>markerSim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Marker simulation</h2>

<h3>Description</h3>

<p>Simulates marker genotypes conditional on the pedigree structure and known
genotypes. Note: This function simulates independent realisations at a single
locus. Equivalently, it can be thought of as independent simulations of
identical, unlinked markers. For simulations of a <em>set</em> of markers, see
<code>profileSim()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">markerSim(
  x,
  N = 1,
  ids = NULL,
  alleles = NULL,
  afreq = NULL,
  mutmod = NULL,
  rate = NULL,
  partialmarker = NULL,
  loopBreakers = NULL,
  seed = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A <code>ped</code> object or a list of such.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>A positive integer: the number of (independent) markers to be
simulated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>A vector containing ID labels of those pedigree members whose
genotypes should be simulated. By default, all individuals are included.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alleles</code></td>
<td>
<p>(Only if <code>partialmarker</code> is NULL.) A vector with allele
labels. If NULL, the following are tried in order:
</p>

<ul>
<li> <p><code>names(afreq)</code>
</p>
</li>
<li>
<p> ‘seq_along(afreq)’
</p>
</li>
<li> <p><code>1:2</code> (Fallback if both <code>alleles</code> and <code>afreq</code> are NULL.)
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>afreq</code></td>
<td>
<p>(Only if <code>partialmarker</code> is NULL.) A numeric vector with allele
frequencies, possibly named with allele labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mutmod, rate</code></td>
<td>
<p>Arguments specifying a mutation model, passed on to
<code>pedtools::marker()</code> (see there for explanations).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partialmarker</code></td>
<td>
<p>Either NULL (resulting in unconditional simulation), a
marker object (on which the simulation should be conditioned) or the name
(or index) of a marker attached to <code>x</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loopBreakers</code></td>
<td>
<p>A numeric containing IDs of individuals to be used as
loop breakers. Relevant only if the pedigree has loops, and only if
<code>partialmarker</code> is non-NULL. See <code>pedtools::breakLoops()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>An integer seed for the random number generator (optional).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This implements (with various time savers) the algorithm used in SLINK of the
LINKAGE/FASTLINK suite. If <code>partialmarker</code> is NULL, genotypes are simulated
by simple gene dropping, using <code>simpleSim()</code>.
</p>


<h3>Value</h3>

<p>A <code>ped</code> object equal to <code>x</code> except its <code>MARKERS</code> entry, which
consists of the <code>N</code> simulated markers.
</p>


<h3>Author(s)</h3>

<p>Magnus Dehli Vigeland
</p>


<h3>References</h3>

<p>G. M. Lathrop, J.-M. Lalouel, C. Julier, and J. Ott, <em>Strategies
for Multilocus Analysis in Humans</em>, PNAS 81(1984), pp. 3443-3446.
</p>


<h3>See Also</h3>

<p><code>profileSim()</code>, <code>simpleSim()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">x = nuclearPed(2)

# Unconditional simulation
markerSim(x, N = 2, alleles = 1:3)

# Conditional on one child being homozygous 1/1
x = addMarker(x, "3" = "1/1", alleles = 1:3)
markerSim(x, N = 2, partialmarker = 1)
markerSim(x, N = 1, ids = 4, partialmarker = 1, verbose = FALSE)

</code></pre>


</div>