<div class="container">

<table style="width: 100%;"><tr>
<td>plot.taxonomy</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot simulated taxonomy</h2>

<h3>Description</h3>

<p>This function is adapted from the <em>ape</em> function <code>plot.phylo</code> used to plot phylogenetic trees.
The function can be used to plot simulated taxonomy along with the corresponding tree.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'taxonomy'
plot(
  x,
  tree,
  show.mode = TRUE,
  show.legend = TRUE,
  legend.position = "bottomleft",
  root.edge = TRUE,
  hide.edge = FALSE,
  edge.width = 1,
  show.tip.label = FALSE,
  align.tip.label = FALSE,
  taxa.palette = "viridis",
  cex = 1.2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Taxonomy object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>Phylo object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.mode</code></td>
<td>
<p>Indicate speciation mode.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.legend</code></td>
<td>
<p>Add a legend for the symbols indicating different speciation modes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend.position</code></td>
<td>
<p>Position of the legend. Options include <code>"bottomleft"</code> (default), <code>"topleft"</code>, <code>"bottomright"</code>, <code>"topright"</code> or a vector of length 2 with x, y coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>root.edge</code></td>
<td>
<p>If TRUE include the root edge (default = TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide.edge</code></td>
<td>
<p>If TRUE hide the root edge but still incorporate it into the automatic timescale (default = FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>edge.width</code></td>
<td>
<p>A numeric vector giving the width of the branches of the plotted phylogeny. These are taken to be in the same order as the component edge of <code>tree</code>. If fewer widths are given than the number of edges, then the values are recycled.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.tip.label</code></td>
<td>
<p>Whether to show the tip labels on the phylogeny (defaults to FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>align.tip.label</code></td>
<td>
<p>A logical value or an integer. If TRUE, the tips are aligned and dotted lines are drawn between the tips of the tree and the labels. If an integer, the tips are aligned and this gives the type of the lines (lty).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxa.palette</code></td>
<td>
<p>Colour palette used for taxa. Colours are assigned to taxa using the function <code>grDevices::hcl.colors()</code> and the default palette is "viridis". Other colour blind friendly palettes include <code>"Blue-Red 3"</code> and <code>"Green-Brown"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>Numeric value giving the factor used to scale the points representing the fossils when <code>show.fossils = TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional parameters to be passed to <code>plot.default</code>.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">set.seed(123)

## simulate tree
t = TreeSim::sim.bd.taxa(8, 1, 1, 0.3)[[1]]

## simulate taxonomy
s = sim.taxonomy(t, 0.5, 1)

## plot the output
plot(s, t)

</code></pre>


</div>