<div class="container">

<table style="width: 100%;"><tr>
<td>plot.fossils</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot simulated fossils</h2>

<h3>Description</h3>

<p>This function is adapted from the <em>ape</em> function <code>plot.phylo</code> used to plot phylogenetic trees.
The function can be used to plot simulated fossils (<code>show.fossils = TRUE</code>), with or without the corresponding tree (<code>show.tree = TRUE</code>),
stratigraphic intervals (<code>show.strata = TRUE</code>), stratigraphic ranges (<code>show.ranges = TRUE</code>) and sampling proxy data (<code>show.proxy = TRUE</code>).
Interval ages can be specified as a vector (<code>interval.ages</code>) or a uniform set of interval ages can be specified using the
number of intervals (<code>strata</code>) and maximum interval age (<code>max</code>), where interval length <code class="reqn">= max.age/strata</code>.
If no maximum age is specified, the function calculates a maximum interval age slightly older than the root edge (or root age if <code>root.edge = FALSE</code>),
using the function <code>tree.max()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'fossils'
plot(
  x,
  tree,
  show.fossils = TRUE,
  show.tree = TRUE,
  show.ranges = FALSE,
  show.strata = FALSE,
  strata = 1,
  max.age = NULL,
  interval.ages = NULL,
  binned = FALSE,
  show.axis = TRUE,
  show.proxy = FALSE,
  proxy.data = NULL,
  show.preferred.environ = FALSE,
  preferred.environ = NULL,
  show.taxonomy = FALSE,
  taxonomy = NULL,
  show.unknown = FALSE,
  rho = 1,
  root.edge = TRUE,
  hide.edge = FALSE,
  edge.width = 1,
  show.tip.label = FALSE,
  align.tip.label = FALSE,
  reconstructed = FALSE,
  fossil.col = 1,
  range.col = rgb(0, 0, 1),
  extant.col = 1,
  taxa.palette = "viridis",
  col.axis = "gray35",
  cex = 1.2,
  pch = 18,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>Fossils object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>Phylo object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.fossils</code></td>
<td>
<p>If TRUE plot fossils (default = TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.tree</code></td>
<td>
<p>If TRUE plot the tree  (default = TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.ranges</code></td>
<td>
<p>If TRUE plot stratigraphic ranges (default = FALSE). If <code>show.taxonomy = FALSE</code> all occurrences along a single edge are grouped together (i.e. function assumes all speciation is symmetric).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.strata</code></td>
<td>
<p>If TRUE plot strata  (default = FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strata</code></td>
<td>
<p>Number of stratigraphic intervals (default = 1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.age</code></td>
<td>
<p>Maximum age of a set of equal length intervals. If no value is specified (<code>max = NULL</code>), the function uses a maximum age based on tree height.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>interval.ages</code></td>
<td>
<p>Vector of stratigraphic interval ages, starting with the minimum age of the youngest interval and ending with the maximum age of the oldest interval.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>binned</code></td>
<td>
<p>If TRUE fossils are plotted at the mid point of each interval.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.axis</code></td>
<td>
<p>If TRUE plot x-axis (default = TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.proxy</code></td>
<td>
<p>If TRUE add profile of sampling data to plot (e.g. rates in time-dependent rates model) (default = FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>proxy.data</code></td>
<td>
<p>Vector of sampling proxy data (default = NULL). Should be as long as the number of stratigraphic intervals.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.preferred.environ</code></td>
<td>
<p>If TRUE add species preferred environmental value (e.g. water depth) (default = FALSE). Only works if combined with <code>show.proxy = TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>preferred.environ</code></td>
<td>
<p>Preferred environmental value (e.g. water depth). Currently only one value can be shown.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.taxonomy</code></td>
<td>
<p>If TRUE highlight species taxonomy.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxonomy</code></td>
<td>
<p>Taxonomy object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.unknown</code></td>
<td>
<p>If TRUE plot fossils with unknown taxonomic affiliation (i.e. sp = NA) (default = FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rho</code></td>
<td>
<p>Extant species sampling probability (default = 1). Will be disregarded if fossils object already contains extant samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>root.edge</code></td>
<td>
<p>If TRUE include the root edge (default = TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide.edge</code></td>
<td>
<p>If TRUE hide the root edge but still incorporate it into the automatic timescale (default = FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>edge.width</code></td>
<td>
<p>A numeric vector giving the width of the branches of the plotted phylogeny. These are taken to be in the same order as the component edge of <code>tree</code>. If fewer widths are given than the number of edges, then the values are recycled.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show.tip.label</code></td>
<td>
<p>Whether to show the tip labels on the phylogeny (defaults to FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>align.tip.label</code></td>
<td>
<p>A logical value or an integer. If TRUE, the tips are aligned and dotted lines are drawn between the tips of the tree and the labels. If an integer, the tips are aligned and this gives the type of the lines (following <code>lty</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reconstructed</code></td>
<td>
<p>If TRUE plot the reconstructed tree. If fossils object contains no extant samples, the function assumes rho = 1 and includes all species at the present.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fossil.col</code></td>
<td>
<p>Colour of fossil occurrences. A vector equal to the length of the fossils object can be used to assign different colours.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>range.col</code></td>
<td>
<p>Colour of stratigraphic ranges.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extant.col</code></td>
<td>
<p>Colour of extant samples. If <code>show.taxonomy = TRUE</code>, <code>extant.col</code> will be ignored.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxa.palette</code></td>
<td>
<p>Colour palette used if <code>show.fossils = TRUE</code>. Colours are assigned to taxa using the function <code>grDevices::hcl.colors()</code> and the default palette is "viridis". Other colour blind friendly palettes include <code>"Blue-Red 3"</code> and <code>"Green-Brown"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.axis</code></td>
<td>
<p>Colour of the time scale axis (default = "gray35").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex</code></td>
<td>
<p>Numeric value giving the factor used to scale the points representing the fossils when <code>show.fossils = TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pch</code></td>
<td>
<p>Numeric value giving the symbol used for the points representing the fossils when <code>show.fossils = TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional parameters to be passed to <code>plot.default</code>.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">set.seed(123)

## simulate tree
t = TreeSim::sim.bd.taxa(8, 1, 1, 0.3)[[1]]

## simulate fossils under a Poisson sampling process
f = sim.fossils.poisson(rate = 3, tree = t)
plot(f, t)
# add a set of equal length strata
plot(f, t, show.strata = TRUE, strata = 4)
# show stratigraphic ranges
plot(f, t, show.strata = TRUE, strata = 4, show.ranges = TRUE)

## simulate fossils and highlight taxonomy
s = sim.taxonomy(t, 0.5, 1)
f = sim.fossils.poisson(rate = 3, taxonomy = s)
plot(f, t, taxonomy = s, show.taxonomy = TRUE, show.ranges = TRUE)


## simulate fossils under a non-uniform model of preservation
# assign a max interval based on tree height
max.age = tree.max(t)
times = c(0, 0.3, 1, max.age)
rates = c(4, 1, 0.1)
f = sim.fossils.intervals(t, interval.ages = times, rates = rates)
plot(f, t, show.strata = TRUE, interval.ages = times)
# add proxy data
plot(f, t, show.strata = TRUE, interval.ages = times, show.proxy = TRUE, proxy.data = rates)

</code></pre>


</div>