<div class="container">

<table style="width: 100%;"><tr>
<td>.knownqpcr_loglike_duo</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Log-likelihood of getting Cq values when exact allele mixing ratios are known, lacking the quartet structure.</h2>

<h3>Description</h3>

<p>Internal function to return the log-likelihood getting the four Cq measurements under true allele frequency for each bulk sample is known. This is a variant for the 'duo' structure dataset and <code>baseChange</code> is absent. <code>obs.housek0</code> and <code>obs.target0</code> are defined for the parity of source code, but they actually accept no data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">.knownqpcr_loglike_duo(X, A, trueY, Digest, Gene, Cq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>A numeric vector that stores the current parameter sizes of <code>meanDNA</code>, <code>targetScale</code>, <code>sdMeasure</code>, <code>zeroAmount</code>, and <code>EPCR</code> in log scale.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>A</code></td>
<td>
<p>Optionally, you can specify relative DNA content between the samples, as a numeric vector having the same length as the Cq data. If present, <code>A</code> must not include missing values. It is the counterpart of the <code>N</code> argument in <code>freqpcr()</code>, whereas an element of <code>A</code> is not restricted to integer. Because the concentration as a whole is also adjusted with the parameter <code>meanDNA</code> (see Value section), <code>A</code> is used exclusively to reflect the relative contents between the sample solutions. Otherwise, <code>A</code> should be left unspecified (the default is 1 for all replicates).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trueY</code></td>
<td>
<p>A numeric vector having the same length as the Cq data. <code>trueY[i]</code> signifies the exact allele frequency in the <em>i</em>th sample. The values must be between 0 and 1, and <code>NA</code> is not allowed. To improve the estimation accuracy, it is better to include the settings y == 0 (pure S solution) and y == 1 (pure R) in your dataset.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A scalar of the log likelihood.
</p>


</div>