<div class="container">

<table style="width: 100%;"><tr>
<td>make_dummy</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate dummy DNA dataset ready for allele-frequency estimation.</h2>

<h3>Description</h3>

<p>The function generates a dummy dataset of typical RED-<code class="reqn">\Delta\Delta</code>Cq analysis. You can directly feed the output of this function to the first argument of <code>sim_dummy()</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">make_dummy(
  rand.seed,
  P,
  K,
  ntrap,
  npertrap,
  scaleDNA = (1/K) * 1e-06,
  targetScale,
  baseChange,
  EPCR,
  zeroAmount,
  sdMeasure,
  diploid = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>rand.seed</code></td>
<td>
<p>Seed for the R built-in random-number-generator.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>P</code></td>
<td>
<p>A numeric between 0 and 1 giving the population allele frequency from which the test samples are generated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>A positive numeric of the gamma shape parameter of the individual DNA yield.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ntrap, npertrap</code></td>
<td>
<p>Scalar specifying the number of bulk samples (<code>ntrap</code>) and the numbers of individuals contained in each bulk sample (<code>npertrap</code>). Currently limited to the cases that all bulk samples have the same sample size: e.g. (4 + 4 + 4) when <code>ntrap = 3</code> and <code>npertrap = 4</code> hold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scaleDNA</code></td>
<td>
<p>Small positive scalar that specifies the scale parameter of the gamma distribution appriximating the DNA yield from (per-haploid) individual. The yield of <code>2*scaleDNA</code> is expected from a diploid. The quantity is determined as the relative amount, in linear scale, to the termination threshold of the real-time PCR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>targetScale</code></td>
<td>
<p>(<code class="reqn">\delta_{T}</code>) The relative template DNA amount of the target gene to the houskeeping gene, given as a positive numeric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>baseChange</code></td>
<td>
<p>(<code class="reqn">\delta_{B}</code>) The change rate in the template DNA quantities after the restriction enzyme digestion (in the RED-<code class="reqn">\Delta\Delta</code>Cq method), given as a positive numeric. This parameter is not used in <code>freqpcr()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EPCR</code></td>
<td>
<p>(<code class="reqn">\eta</code>) Amplification efficiency per PCR cycle, given as a positive numeric. When <code>EPCR = 1</code>, template DNA doubles every cycle (<code>EPCR + 1 = 2</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>zeroAmount</code></td>
<td>
<p>A numeric between 0 and 1, usually near 0, giving the residue rate of restriction enzyme digestion in RED-<code class="reqn">\Delta\Delta</code>Cq method.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sdMeasure</code></td>
<td>
<p>(<code class="reqn">\sigma_{c}</code>) Scalar. The measurement error (standard deviation) on each Cq value following Normal(0, <code class="reqn">\sigma_{c}^2</code>). If known, given as a positive numeric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diploid</code></td>
<td>
<p>Is the target organism diploidy? Default is <code>FALSE</code>, assuming haploidy. Current implementation of diploidy assumes i.i.d. between the amounts of R and S chromosomes owned by a heterozygote individual, which is unlikely in many animals but necessary for the calculation in a realistic time.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Object of the S4 class CqList, storing the dummy experiment data of Cq-based qPCR analysis. Note that a CqList object in no way contains original information on <code>P</code>, <code>K</code>, <code>targetScale</code>, <code>sdMeasure</code>, and <code>EPCR</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">P &lt;- 0.25
# Just a test: segregation ratios for six bulk samples, 1000 individuals for each.
rmultinom(n=6, size=1000, prob=c(P, 1-P)) # haploidy
rmultinom(6, size=1000, prob=c(P^2, 2*P*(1-P), (1-P)^2)) # diploidy

# Dummy Cq dataset with six bulk samples, each of which comprises eight haploids.
dmy_cq &lt;- make_dummy( rand.seed=1, P=0.15, K=2, ntrap=6, npertrap=8,
                      scaleDNA=1e-07, targetScale=1.5, baseChange=0.3, EPCR=0.95,
                      zeroAmount=1e-3, sdMeasure=0.3, diploid=FALSE )
print(dmy_cq)
</code></pre>


</div>