<div class="container">

<table style="width: 100%;"><tr>
<td>fusions.process</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
LD-RTPCR fusion identification by Sanger
</h2>

<h3>Description</h3>

<p>Automatically interpret gene fusions found by Sanger sequencing using the Ligation-Dependent PCR protocol. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">  fusions.process(input, design, output, sheet = NA, cores = NA, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>

<p>Single character value, the path to the directory containing the AB1 files to process.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>design</code></td>
<td>

<p>Data.frame describing all possible fusions (see Details).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>

<p>Single character value, the path to a directory in which to produce output files (will be created if doesn't yet exists).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sheet</code></td>
<td>

<p>Single character value, the name and path of a CSV file describing the files to process. 3 columns are expected: <code>ID</code> which gives a simpler sample name to use in outputs, <code>way</code> which defines if sequencing was 'forward' or 'reverse', and <code>file</code> which gives the file name and path relative to the <code>input</code> argument.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>

<p>Single integer value, the amount of CPUs to use on the local machine to parallelize the computation. If <code>NA</code>, a guess will be made. If 1, computation will not use the <code>parallel</code> package at all but only loop over samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Further arguments are passed to <code>fusions.process.core</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>design</code> must contain one row for each possible combination of a left primer with a right primer, whether this fusion is expected and relevant or not.
</p>
<p>Expected columns in <code>design</code> are (excluding extra columns required with <code>extra</code>) :
</p>

<dl>
<dt>left.name</dt>
<dd>
<p>Character, the name of the left primer.</p>
</dd>
<dt>left.seq</dt>
<dd>
<p>Character (uppercase), the sequence of the left primer (gene-specific part only).</p>
</dd>
<dt>left.unileft</dt>
<dd>
<p>Character (uppercase), the sequence of the left universal primer used for amplification.</p>
</dd>
<dt>left.symbol</dt>
<dd>
<p>Character, the symbol of the gene targeted by the left primer.</p>
</dd>
<dt>left.GRCh38</dt>
<dd>
<p>Character, the genomic coordinates of the last base of the left primer (chromosome:position:strand).</p>
</dd>
<dt>left.GRCh38_band</dt>
<dd>
<p>Character, the cytogenetic location of the gene targeted by the left primer.</p>
</dd>
<dt>right.name</dt>
<dd>
<p>Character, the name of the right primer.</p>
</dd>
<dt>right.seq</dt>
<dd>
<p>Character (uppercase), the sequence of the right primer (gene-specific part only).</p>
</dd>
<dt>right.uniright</dt>
<dd>
<p>Character (uppercase), the sequence of the right universal primer used for amplification.</p>
</dd>
<dt>right.symbol</dt>
<dd>
<p>Character, the symbol of the gene targeted by the right primer.</p>
</dd>
<dt>right.GRCh38</dt>
<dd>
<p>Character, the genomic coordinates of the last base of the right primer (chromosome:position:strand).</p>
</dd>
<dt>right.GRCh38_band</dt>
<dd>
<p>Character, the cytogenetic location of the gene targeted by the right primer.</p>
</dd>
<dt>seq_forward</dt>
<dd>
<p>Character (uppercase), the complete sequence expected in forward sequencing (concatenation of <code>left.unileft</code>, <code>left.seq</code>, <code>right.seq</code>, <code>right.uniright</code> and the right tail, if any).</p>
</dd>
<dt>seq_reverse</dt>
<dd>
<p>Character (uppercase), the complete sequence expected in reverse sequencing (reverse complement of a concatenation of the left tail, if any, <code>left.unileft</code>, <code>left.seq</code>, <code>right.seq</code>, <code>right.uniright</code>).</p>
</dd>
</dl>
<p>Please contact the authors to obtain a relevant design object.
</p>


<h3>Value</h3>

<p>Invisibly returns the aggregated table of top results for all samples.
</p>
<p>Various files are produced, in location set by the <code>output</code> argument :
</p>

<dl>
<dt>Top.csv</dt>
<dd>
<p>The aggregated table of top results for all samples.</p>
</dd>
<dt>*.pdf</dt>
<dd>
<p>One plot for each sample, showing the sequencing profile and the best alignments found.</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>See Also</h3>

<p><code>generic.process</code>
</p>


</div>