<div class="container">

<table style="width: 100%;"><tr>
<td>classify</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Apply the binary predictor to FSA peaks
</h2>

<h3>Description</h3>

<p>Predict to which class the sample is most likely to belong, using a modified LPS model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  classify(x, plot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>A <code>fsa</code> object with <code>peaks</code> and <code>model</code> attributes.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot</code></td>
<td>

<p>Single logical value, whether to plot a visual representation of the prediction or not.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a list :
</p>
<table>
<tr style="vertical-align: top;">
<td><code>score</code></td>
<td>
<p>The raw score used to make the prediction.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>The probability to belong to each of the two groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>class</code></td>
<td>
<p>The final prediction, as a group name. May be <code>NA</code> if no probability passes the model threshold.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Sylvain Mareschal
</p>


<h3>References</h3>

<p>Radmacher MD, McShane LM, Simon R. <cite>A paradigm for class prediction using gene expression profiles.</cite> J Comput Biol. 2002;9(3):505-11.
</p>
<p>Wright G, Tan B, Rosenwald A, Hurt EH, Wiestner A, Staudt LM. <cite>A gene expression-based method to diagnose clinically distinct subgroups of diffuse large B cell lymphoma.</cite> Proc Natl Acad Sci U S A. 2003 Aug 19;100(17):9991-6.
</p>
<p>Bohers E, Mareschal S, Bouzelfen A, Marchand V, Ruminy P, Maingonnat C, Menard AL, Etancelin P, Bertrand P, Dubois S, Alcantara M, Bastard C, Tilly H, Jardin F. <cite>Targetable activating mutations are very frequent in GCB and ABC diffuse large B-cell lymphoma.</cite> Genes Chromosomes Cancer. 2014 Feb;53(2):144-53.  
</p>


<h3>See Also</h3>

<p><code>read.fsa</code>, <code>peaks.fsa</code>, <code>add.model</code>, <code>generic.process</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  # Example FSA file provided
  fsa &lt;- read.fsa(system.file("extdata/fsa_GEP/A5918.fsa", package="FSAtools"))
  fsa &lt;- align.fsa(fsa)
  
  # Add model from design file
  design &lt;- designFile(system.file("extdata/design_GEP.conf", package="FSAtools"))
  fsa &lt;- add.model(fsa, model=design$GLOBALS$MODEL)
  
  # Add peak heights
  fsa &lt;- peaks.fsa(fsa, peaks=design$GLOBALS$PEAKS)
  
  # Classify sample
  fsa &lt;- classify(fsa, plot=TRUE)
  print(attr(fsa, "classification"))
</code></pre>


</div>