<div class="container">

<table style="width: 100%;"><tr>
<td>FST.GeneSet.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Test the association between an quantitative/dichotomous outcome variable and a large gene-set by a score type test allowing for multiple functional annotation scores.
</h2>

<h3>Description</h3>

<p>Once the preliminary work is done using "FST.prelim()", this function tests a specifc gene.
</p>


<h3>Usage</h3>

<pre><code class="language-R">FST.GeneSet.test(result.prelim,G,Z,GeneSetID,Gsub.id=NULL,weights=NULL,
B=5000,impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>result.prelim</code></td>
<td>

<p>The output of function "FST.prelim()"
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>

<p>Genetic variants in the target gene, an n*p matrix where n is the subject ID and p is the total number of genetic variables. Note that the number of rows in G should be same as the number of subjects. ***The column name should be the variable name, in order to be matched with the GeneSetID.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>

<p>Functional annotation scores, an p*q matrix where p is the total number of genetic variables and q is the number of functional annotation scores. Note that the first column in Z should be all 1 if the users want the original weights of SKAT/burden test to be included.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GeneSetID</code></td>
<td>

<p>A p*2 matrix indicating the genes in which the variables are located, where the first column is the genes' name and the second column is the variables' name.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an n dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weights</code></td>
<td>

<p>A numeric vector of weights for genetic variants (The length should be same as the number of genetic variants in the set). These weights are usually based on minor allele frequencies. The default is NULL, where the beta(1,25) weights are applied.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be "random", "fixed" or "bestguess". Given the estimated allele frequency, "random" simulates the genotype from binomial distribution; "fixed" uses the genotype expectation; "Best guess" uses the genotype with highest probability.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.marker</code></td>
<td>

<p>number of heterozygous SNPs in the SNP set.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## FST.prelim does the preliminary data management.
# Input: Y, X (covariates)
## FST.test tests a region.
# Input: G (genetic variants), Z (functional annotation scores) and result of FST.prelim

library(FSTpackage)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by d matrix
# G: genotype matrix, n by p matrix where n is the total number of subjects
# Z: functional annotation matrix, p by q matrix

data(FST.example)
Y&lt;-FST.example$Y;X&lt;-FST.example$X;G&lt;-FST.example$G;Z&lt;-FST.example$Z;GeneSetID&lt;-FST.example$GeneSetID

# Preliminary data management
result.prelim&lt;-FST.prelim(Y,X=X,out_type='D')

# test with 5000 bootstrap replicates
result&lt;-FST.GeneSet.test(result.prelim,G,Z,GeneSetID,B=5000)
</code></pre>


</div>