<div class="container">

<table style="width: 100%;"><tr>
<td>PseEIIP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pseudo Electron-Ion Interaction Pseudopotentials of Trinucleotide (PseEIIP)</h2>

<h3>Description</h3>

<p>This function calculates the pseudo electron-ion interaction for each sequence.
It creates a feature vector for each sequence.
The vector contains a value for each for each tri-nucleotide.
The value is computed by multiplying the aggregate value of electron-ion interaction of each nucleotide
</p>


<h3>Usage</h3>

<pre><code class="language-R">PseEIIP(seqs, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns a feature matrix which the number of rows is equal to the number of sequences and the
number of columns is 4^3=64.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. "iLearn: an integrated platform and meta-learner for feature engineering, machine-learning analysis and modeling of DNA, RNA and protein sequence data." Briefings in bioinformatics 21.3 (2020): 1047-1057.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-PseEIIP(seqs = LNC50Nuc)
</code></pre>


</div>