<div class="container">

<table style="width: 100%;"><tr>
<td>AESNN3</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Learn from alignments (AESNN3)</h2>

<h3>Description</h3>

<p>This function replace each amino acid of the sequence with a three-dimensional vector. Values are taken from the three hidden units of the neural network trained on structure alignments.
The AESNN3 function can be applied to encode peptides of equal length.
</p>


<h3>Usage</h3>

<pre><code class="language-R">AESNN3(seqs, label = c(), outFormat = "mat", outputFileDist = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*(5)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat parameter for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Lin K, May AC, Taylor WR. Amino acid encoding schemes from protein structure alignments: multi-dimensional vectors to describe residue types. J Theor Biol (2002).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat&lt;-AESNN3(seqs = ptmSeqsVect,outFormat="mat")
</code></pre>


</div>