<div class="container">

<table style="width: 100%;"><tr>
<td>KNNProtein</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>K-Nearest Neighbor for Protein (KNNProtein)</h2>

<h3>Description</h3>

<p>This function is like KNNPeptide with the difference that similarity score is computed by Needleman-Wunsch algorithm.
</p>


<h3>Usage</h3>

<pre><code class="language-R">KNNProtein(seqs, trainSeq, percent = 30, labeltr = c(), label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a fasta file with amino acids sequences. Each sequence starts
with a '&gt;' character. Also it could be a string vector such that each element is a protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trainSeq</code></td>
<td>
<p>is a fasta file with amino acids sequences. Each sequence starts
with a '&gt;' character. Also it could be a string vector such that each element is a protein sequence. Eaxh sequence in the training set
is associated with a label. The label is found in the parameret labeltr.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>percent</code></td>
<td>
<p>determines the threshold which is used to identify sequences (in the training set) which are similar to the input sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labeltr</code></td>
<td>
<p>This parameter is a vector whose length is equivalent to the number of sequences in the training set. It shows class of
each sequence in the trainig set.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns a feature matrix such that number of columns is number of classes multiplied by percent and number of rows is equal to the number of the sequences.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. "iFeature: a python package and web server for features extraction and selection from protein and peptide sequences." Bioinformatics 34.14 (2018): 2499-2502.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
ptmSeqsVect&lt;-ptmSeqsVect[1:2]
ptmSeqsVect&lt;-sapply(ptmSeqsVect,function(seq){substr(seq,1,31)})

posSeqs&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/poSeqPTM101.csv"))[,2])
negSeqs&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/negSeqPTM101.csv"))[,2])

posSeqs&lt;-posSeqs[1:3]
negSeqs&lt;-negSeqs[1:3]

posSeqs&lt;-sapply(posSeqs,function(seq){substr(seq,1,31)})
negSeqs&lt;-sapply(negSeqs,function(seq){substr(seq,1,31)})

trainSeq&lt;-c(posSeqs,negSeqs)

labelPos&lt;-rep(1,length(posSeqs))
labelNeg&lt;-rep(0,length(negSeqs))

labeltr&lt;-c(labelPos,labelNeg)

mat&lt;-KNNProtein(seqs=ptmSeqsVect,trainSeq=trainSeq,percent=5,labeltr=labeltr)
</code></pre>


</div>