<div class="container">

<table style="width: 100%;"><tr>
<td>SAAC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Splitted Amino Acid Composition (SAAC)</h2>

<h3>Description</h3>

<p>This function splits the input sequence into three parts. The first part is N-terminal
and the third part is C-terminal and middle part contains all amino acids between these two part.
N-terminal will be determined by the first numNterm amino acid in the sequences and C-terminal is
determined by numCterm of the last amino acids in the sequence. Users should enter numNterm and numCterm
parameters. Their default value is 25.
The function calculates kAAComposition for each of the three parts.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SAAC(seqs, k = 1, numNterm = 5, numCterm = 5, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>shows which type of amino acid composition applies to the parts.
For example, the amino acid composition is applied when k=1 and when k=2, the dipeptide Composition is applied.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numNterm</code></td>
<td>
<p>shows how many amino acids should be considered for N-terminal.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numCterm</code></td>
<td>
<p>shows how many amino acids should be considered for C-terminal.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>It returns a feature matrix. The number of rows is equal to the number of sequences.
The number of columns is (3*(20^k)).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-SAAC(seqs=filePrs,k=1,numNterm=15,numCterm=15)
</code></pre>


</div>