<div class="container">

<table style="width: 100%;"><tr>
<td>PSEAAC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pseudo-Amino Acid Composition (Parallel) (PSEAAC)</h2>

<h3>Description</h3>

<p>This function calculates the pseudo amino acid composition (parallel)
for each sequence.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PSEAAC(
  seqs,
  aaIDX = c("ARGP820101", "HOPT810101", "Mass"),
  lambda = 30,
  w = 0.05,
  l = 1,
  threshold = 1,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aaIDX</code></td>
<td>
<p>is a vector of Ids or indexes of the user-selected physicochemical properties in the aaIndex2 database.
The default values of the vector are the hydrophobicity ids and hydrophilicity ids and Mass of residual
in the amino acid index file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda</code></td>
<td>
<p>is a tuning parameter. Its value indicates the maximum number of spaces between amino acid pairs. The number
changes from 1 to lambda.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>w</code></td>
<td>
<p>(weight) is a tuning parameter. It changes in from 0 to 1. The default value is 0.05.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>l</code></td>
<td>
<p>This parameter keeps the value of l in lmer composition. The lmers form the first 20^l elements of the APAAC descriptor.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>is a number between (0 , 1]. It deletes aaIndexes which have a correlation
bigger than the threshold. The default value is 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A feature matrix such that the number of columns is 20^l+(lambda) and the number of rows is equal to the number of sequences.
</p>


<h3>Examples</h3>

<pre><code class="language-R">filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-PSEAAC(seqs=filePrs,l=2)
</code></pre>


</div>