<div class="container">

<table style="width: 100%;"><tr>
<td>EGAAComposition</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Enhanced Grouped Amino Acid Composition (EGAAComposition)</h2>

<h3>Description</h3>

<p>In this function, amino acids are first grouped into user-defined categories. Then, enhanced grouped amino acid composition is computed.
For details about the enhanced feature, please refer to function EAAComposition.
Please note that this function differs from function EAAComposition which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class="language-R">EGAAComposition(
  seqs,
  winSize = 50,
  overLap = TRUE,
  Grp = "locFus",
  label = c(),
  outFormat = "mat",
  outputFileDist = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>winSize</code></td>
<td>
<p>shows the size of sliding window. It is a numeric value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overLap</code></td>
<td>
<p>This parameter shows how the window
moves on the sequence. If the overlap is set to TRUE, the next window would have distance 1 with
the previous window. Otherwise, the next window will start from the next amino acid after the previous window.
There is no overlap between the next and previous windows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is ((number of categorizes) * (number of windows))
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
dir = tempdir()
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
ptmSeqsVect&lt;-as.vector(read.csv(paste0(ptmSeqsADR,"/ptmVect101AA.csv"))[,2])
mat1&lt;-EGAAComposition(seqs = ptmSeqsVect,winSize=20,overLap=FALSE,Grp="locFus")

mat2&lt;-EGAAComposition(seqs = ptmSeqsVect,winSize=30,overLap=FALSE,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")),outFormat="mat")

ad&lt;-paste0(dir,"/EGrpaaCompos.txt")
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
EGAAComposition(seqs = filePrs,winSize=20,Grp="cTriad",outFormat="txt"
,outputFileDist=ad)


unlink("dir", recursive = TRUE)
</code></pre>


</div>