<div class="container">

<table style="width: 100%;"><tr>
<td>ANF_DNA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Accumulated Nucleotide Frequency (ANF_DNA)</h2>

<h3>Description</h3>

<p>This function replaces nucleotides with a four-length vector.
The first three elements represent the nucleotides and
the forth holds the frequency of the nucleotide from the beginning of the sequence until the position of the nucleotide in the sequence.
'A' will be replaced with c(1, 1, 1, freq), 'C' with c(0, 1, 0, freq),'G' with c(1, 0, 0, freq), and 'T' with c(0, 0, 1, freq).
</p>


<h3>Usage</h3>

<pre><code class="language-R">ANF_DNA(seqs, outFormat = "mat", outputFileDist = "", label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outFormat</code></td>
<td>
<p>(output format) can take two values: 'mat'(matrix) and 'txt'. The default value is 'mat'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputFileDist</code></td>
<td>
<p>shows the path and name of the 'txt' output file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The output depends on the outFormat parameter which can be either 'mat' or 'txt'. If outFormat is 'mat', the function returns a feature
matrix for sequences with the same length such that the number of columns is (sequence length)*(4)
and the number of rows is equal to the number of sequences.
If the outFormat is 'txt', the output is written to a tab-delimited file.
</p>


<h3>Note</h3>

<p>This function is provided for sequences with the same lengths.
Users can use 'txt' option in outFormat for sequences with different lengths.
Warning: If outFormat is set to 'mat' for sequences with different lengths, it returns an error.
Also, when output format is 'txt', label information is not shown in the text file.
It is noteworthy that 'txt' format is not usable for machine learning purposes if sequences have different sizes. Otherwise 'txt' format
is also usable for machine learning purposes.
</p>


<h3>References</h3>

<p>Chen, W., Tran, H., Liang, Z. et al. Identification and analysis of the N6-methyladenosine in the Saccharomyces cerevisiae transcriptome. Sci Rep 5, 13859 (2015).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
LNCSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")
LNC50Nuc&lt;-as.vector(read.csv(paste0(LNCSeqsADR,"/LNC50Nuc.csv"))[,2])
mat&lt;-ANF_DNA(seqs = LNC50Nuc,outFormat="mat")


</code></pre>


</div>