<div class="container">

<table style="width: 100%;"><tr>
<td>PseKRAAC_T16</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Pseudo K_tuple Reduced Amino Acid Composition Type-16 (PseKRAAC_T16)</h2>

<h3>Description</h3>

<p>There are 16 types of PseKRAAC function.
In the functions, a (user-selected) grouping of the amino acids might be used to reduce the amino acid alphabet.
Also, the functions have a type parameter.
The parameter determines the protein sequence analyses which can be either gap or lambda-correlation.
PseKRAAC_type16(PseKRAAC_T16) contains Grp 2-16,20.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PseKRAAC_T16(
  seqs,
  type = "gap",
  Grp = 2,
  GapOrLambdaValue = 2,
  k = 4,
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>This parameter has two valid value "lambda" and "gap". "lambda"
calls lambda_model function and "gap" calls gap_model function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Grp</code></td>
<td>
<p>is a numeric value. It shows the id of an amino acid group.
Please find the available groups in the detail section.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GapOrLambdaValue</code></td>
<td>
<p>is an integer.
If type is gap, this value shows number of gaps between two k-mers.
If type is lambda, the value of GapOrLambdaValue shows the number of gaps between each two amino acids of k-mers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>This parameter keeps the value of k in k-mer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Groups:
</p>
<p>2=c('IMVLFWY', 'GPCASTNHQEDRK'),
3=c('IMVLFWY', 'GPCAST', 'NHQEDRK'),
4=c('IMVLFWY', 'G', 'PCAST', 'NHQEDRK'),
5=c('IMVL', 'FWY', 'G', 'PCAST', 'NHQEDRK'),
6=c('IMVL', 'FWY', 'G', 'P', 'CAST', 'NHQEDRK'),
7=c('IMVL', 'FWY', 'G', 'P', 'CAST', 'NHQED', 'RK'),
8=c('IMV', 'L', 'FWY', 'G', 'P', 'CAST', 'NHQED', 'RK'),
9=c('IMV', 'L', 'FWY', 'G', 'P', 'C', 'AST', 'NHQED', 'RK'),
10=c('IMV', 'L', 'FWY', 'G', 'P', 'C', 'A', 'STNH', 'RKQE', 'D'),
11=c('IMV', 'L', 'FWY', 'G', 'P', 'C', 'A', 'STNH', 'RKQ', 'E', 'D'),
12=c('IMV', 'L', 'FWY', 'G', 'P', 'C', 'A', 'ST', 'N', 'HRKQ', 'E', 'D'),
13=c('IMV', 'L', 'F', 'WY', 'G', 'P', 'C', 'A', 'ST', 'N', 'HRKQ', 'E', 'D'),
14=c('IMV', 'L', 'F', 'WY', 'G', 'P', 'C', 'A', 'S', 'T', 'N', 'HRKQ', 'E', 'D'),
15=c('IMV', 'L', 'F', 'WY', 'G', 'P', 'C', 'A', 'S', 'T', 'N', 'H', 'RKQ', 'E', 'D'),
16=c('IMV', 'L', 'F', 'W', 'Y', 'G', 'P', 'C', 'A', 'S', 'T', 'N', 'H', 'RKQ', 'E', 'D'),
20=c('I', 'M', 'V', 'L', 'F', 'W', 'Y', 'G', 'P', 'C', 'A', 'S', 'T', 'N', 'H', 'R', 'K', 'Q', 'E', 'D')
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (Grp)^k.
</p>


<h3>References</h3>

<p>Zuo, Yongchun, et al. "PseKRAAC: a flexible web server for generating pseudo K-tuple reduced amino acids composition." Bioinformatics 33.1 (2017): 122-124.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-PseKRAAC_T16(seqs=filePrs,type="gap",Grp=4,GapOrLambdaValue=3,k=2)

mat2&lt;-PseKRAAC_T16(seqs=filePrs,type="lambda",Grp=4,GapOrLambdaValue=3,k=2)
</code></pre>


</div>