<div class="container">

<table style="width: 100%;"><tr>
<td>Mismatch_RNA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Mismatch_RNA (Mismatch_RNA)</h2>

<h3>Description</h3>

<p>This function also calculates the frequencies of all k-mers in the sequence but alows maximum m mismatch.
m&lt;k.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Mismatch_RNA(seqs, k = 3, m = 2, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file containing ribonucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a ribonucleotide sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>This parameter can be a number which shows kmer.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m</code></td>
<td>
<p>This parametr shows muximum number of mismatches.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns depends on the rng vector. For each value k in the vector, (4)^k columns are created in the matrix.
</p>


<h3>References</h3>

<p>Liu, B., Gao, X. and Zhang, H. BioSeq-Analysis2.0: an updated platform for analyzing DNA, RNA and protein sequences at sequence level and residue level based on machine learning approaches. Nucleic Acids Res (2019).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
fileLNC&lt;-system.file("extdata/Carica_papaya101RNA.txt",package="ftrCOOL")
mat&lt;-Mismatch_RNA(seqs=fileLNC)


</code></pre>


</div>