<div class="container">

<table style="width: 100%;"><tr>
<td>AAKpartComposition</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Amino Acid to K Part Composition (AAKpartComposition)</h2>

<h3>Description</h3>

<p>In this function, each sequence is divided into k equal partitions.
The length of each part is equal to ceiling(l(lenght of the sequence)/k).
The last part can have a different length containing the residual amino acids.
The amino acid composition is calculated for each part.
</p>


<h3>Usage</h3>

<pre><code class="language-R">AAKpartComposition(seqs, k = 3, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>is an integer value. Each sequence should be divided to k partition(s).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a feature matrix with k*20 number of columns. The number of rows is equal to the number of
sequences.
</p>


<h3>Note</h3>

<p>Warning: The length of all sequences should be greater than k.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat&lt;-AAKpartComposition(seqs=filePrs,k=5,normalized=FALSE)

</code></pre>


</div>