<div class="container">

<table style="width: 100%;"><tr>
<td>ExpectedValueGAA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Expected Value for Grouped Amino Acid (ExpectedValueGAA)</h2>

<h3>Description</h3>

<p>This function is introduced by this package for the first time.
In this function, amino acids are first grouped into user-defined categories.
Later, the expected value of grouped amino acids is computed.
Please note that this function differs from Function ExpectedValueAA which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ExpectedValueGAA(seqs, k = 3, Grp = "locFus", normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>is an integer value. The default is three.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>for more information about ExpectedValueGAA, please refer to function ExpectedValueKmer.
</p>


<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns is (number of categories)^k.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-ExpectedValueGAA(seqs=filePrs,k=2,Grp="locFus")

mat2&lt;-ExpectedValueGAA(seqs=filePrs,k=1,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")))
</code></pre>


</div>