<div class="container">

<table style="width: 100%;"><tr>
<td>conjointTriad</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Conjoint Triad (conjointTriad)</h2>

<h3>Description</h3>

<p>This function calculates the grouped tripeptide composition with the conjoint triad grouping type.
</p>


<h3>Usage</h3>

<pre><code class="language-R">conjointTriad(seqs, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows equals to the number of sequences and
the number of columns is 7^3.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-conjointTriad(seqs=filePrs)

</code></pre>


</div>