<div class="container">

<table style="width: 100%;"><tr>
<td>PSTNPds</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Position-Specific Trinucleotide Propensity based on double-strand (PSTNPds)</h2>

<h3>Description</h3>

<p>This function works like PSTNPss_DNA except that it considers T as A and G as C. So it
converts Ts in the sequence to A and Gs to C. Then, it works with 2 alphabets A and C.
For more details refer to PSTNPss_DNA.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PSTNPds(seqs, pos, neg, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>is a fasta file containing nucleotide sequences. Each sequence starts
with '&gt;'. Also, the value of this parameter can be a string vector.
The sequences are positive sequences in the training model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>neg</code></td>
<td>
<p>is a fasta file containing nucleotide sequences. Each sequence starts
with '&gt;'. Also, the value of this parameter can be a string vector.
The sequences are negative sequences in the training model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equal to the number of sequences.
It shows the class of each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>It returns a feature matrix. The number of columns is equal to the length of sequences minus two
and the number of rows is equal to the number of sequences.
</p>


<h3>Note</h3>

<p>The length of the sequences in positive and negative data sets and the input sets
should be equal.
</p>


<h3>References</h3>

<p>Chen, Zhen, et al. "iLearn: an integrated platform and meta-learner for feature engineering, machine-learning analysis and modeling of DNA, RNA and protein sequence data." Briefings in bioinformatics 21.3 (2020): 1047-1057.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
ptmSeqsADR&lt;-system.file("extdata/",package="ftrCOOL")

posSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/posData.txt"),alphabet="dna")
negSeqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/negData.txt"),alphabet="dna")
seqs&lt;-fa.read(file=paste0(ptmSeqsADR,"/testData.txt"),alphabet="dna")


PSTNPds(seqs=seqs,pos=posSeqs[1],neg=negSeqs[1])



</code></pre>


</div>