<div class="container">

<table style="width: 100%;"><tr>
<td>MMI_DNA</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Multivariate Mutual Information_DNA (MMI_DNA)</h2>

<h3>Description</h3>

<p>MMI computes mutual information based on 2-mers T2 =  AA, AC, AG, AT, CC, CG, CT, GG, GT, TT
and 3-mers T3 = AAA, AAC, AAG, AAT, ACC, ACG, ACT, AGG, AGT, ATT, CCC, CCG, CCT, CGG, CGT, CTT, GGG, GGT, GTT and TTT
for more information please check the reference part.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MMI_DNA(seqs, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file containing nucleotide sequences. The sequences start
with '&gt;'. Also, seqs could be a string vector. Each element of the vector is a nucleotide sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>It is a feature matrix. The number of columns is 30 and the number of rows is equal to the number of sequences.
</p>


<h3>References</h3>

<p>Zhen Chen, Pei Zhao, Chen Li, Fuyi Li, Dongxu Xiang, Yong-Zi Chen, Tatsuya Akutsu, Roger J Daly, Geoffrey I Webb, Quanzhi Zhao, Lukasz Kurgan, Jiangning Song. iLearnPlus: a comprehensive and automated machine-learning platform for nucleic acid and protein sequence analysis, prediction and visualization, Nucleic Acids Research (2021).
</p>


<h3>Examples</h3>

<pre><code class="language-R">
fileLNC&lt;-system.file("extdata/Athaliana_LNCRNA.fa",package="ftrCOOL")
mat&lt;-MMI_DNA(seqs=fileLNC)
</code></pre>


</div>