<div class="container">

<table style="width: 100%;"><tr>
<td>CTD</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Composition_Transition_Distribution (CTD)</h2>

<h3>Description</h3>

<p>This function calculates the composition, transition, and distribution for each sequence.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CTD(seqs, normalized = FALSE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Output is a combination of three different matrices: Composition, Transition, and Distribution.
You can obtain any of the three matrices by executing the corresponding function, i.e., CTDC, CTDT, and CTDD.
</p>


<h3>References</h3>

<p>Dubchak, Inna, et al. "Prediction of protein folding class using global description of amino acid sequence." Proceedings of the National Academy of Sciences 92.19 (1995): 8700-8704.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
CTDtotal&lt;-CTD(seqs=filePrs,normalized=FALSE)

</code></pre>


</div>