<div class="container">

<table style="width: 100%;"><tr>
<td>kAAComposition</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>k Amino Acid Composition (kAAComposition)</h2>

<h3>Description</h3>

<p>This function calculates the frequency of all k-mers in the sequence(s).
</p>


<h3>Usage</h3>

<pre><code class="language-R">kAAComposition(seqs, rng = 3, upto = FALSE, normalized = TRUE, label = c())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rng</code></td>
<td>
<p>This parameter can be a number or a vector. Each entry of the vector holds the value of k in the k-mer composition.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>upto</code></td>
<td>
<p>It is a logical parameter. The default value is FALSE. If rng is a number and upto is set to TRUE, rng is converted
to a vector with values from 1 to rng.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>This function returns a feature matrix. The number of rows is equal to the number of sequences and
the number of columns depends on rng vector. For each value k in the vector, (20)^k columns are created in the matrix.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")

mat1&lt;-kAAComposition(seqs=filePrs,rng=3,upto=TRUE)
mat2&lt;-kAAComposition(seqs=filePrs,rng=c(1,3),upto=TRUE)
</code></pre>


</div>