<div class="container">

<table style="width: 100%;"><tr>
<td>GAAKpartComposition</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Grouped Amino Acid K Part Composition (GAAKpartComposition)</h2>

<h3>Description</h3>

<p>In this function, amino acids are first grouped into user-defined categories.
Later, the composition of the grouped amino acid k part is computed.
Please note that this function differs from AAKpartComposition which works on individual amino acids.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GAAKpartComposition(
  seqs,
  k = 5,
  normalized = TRUE,
  Grp = "locFus",
  label = c()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p>is a FASTA file with amino acid sequences. Each sequence starts
with a '&gt;' character. Also, seqs could be a string vector. Each element of the vector is a peptide/protein sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>is an integer. Each sequence should be divided to k partition(s).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalized</code></td>
<td>
<p>is a logical parameter. When it is FALSE, the return value of the function does not change. Otherwise, the return value is normalized using the length of the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Grp</code></td>
<td>
<p>is a list of vectors containig amino acids. Each vector represents a category. Users can define a customized amino acid grouping, provided that the sum of all amino acids is 20 and there is no repeated amino acid in the groups.
Also, users can choose 'cTriad'(conjointTriad), 'locFus', or 'aromatic'. Each option provides specific information about the type of an amino acid grouping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label</code></td>
<td>
<p>is an optional parameter. It is a vector whose length is equivalent to the number of sequences. It shows the class of
each entry (i.e., sequence).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a feature matrix with k*(number of categorizes) number of columns. The number of rows is equal to the number of
sequences.
</p>


<h3>Note</h3>

<p>Warning: The length of all sequences should be greater than k.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
filePrs&lt;-system.file("extdata/proteins.fasta",package="ftrCOOL")
mat1&lt;-GAAKpartComposition(seqs=filePrs,k=5,Grp="aromatic")

mat2&lt;-GAAKpartComposition(seqs=filePrs,k=3,normalized=FALSE,Grp=
list(Grp1=c("G","A","V","L","M","I","F","Y","W"),Grp2=c("K","R","H","D","E")
,Grp3=c("S","T","C","P","N","Q")))
</code></pre>


</div>